diff -Naur micropython-1.15/ports/unix/main.c micropython-phoenix/ports/unix/main.c
--- micropython-1.15/ports/unix/main.c	2021-04-18 16:27:07.000000000 +0200
+++ micropython-phoenix/ports/unix/main.c	2021-06-30 11:48:28.980936563 +0200
@@ -37,6 +37,7 @@
 #include <sys/types.h>
 #include <errno.h>
 #include <signal.h>
+#include <limits.h>
 
 #include "py/compile.h"
 #include "py/runtime.h"
@@ -59,7 +98,7 @@
 #if MICROPY_ENABLE_GC
 // Heap size of GC heap (if enabled)
 // Make it larger on a 64 bit machine, because pointers are larger.
-long heap_size = 1024 * 1024 * (sizeof(mp_uint_t) / 4);
+long heap_size = UPYTH_HEAPSZ * (sizeof(mp_uint_t) / 4);
 #endif
 
 STATIC void stderr_print_strn(void *env, const char *str, size_t len) {
@@ -451,7 +490,7 @@
     signal(SIGPIPE, SIG_IGN);
     #endif
 
-    mp_stack_set_limit(40000 * (sizeof(void *) / 4));
+    mp_stack_set_limit(UPYTH_STACKSZ * (sizeof(void *) / 4));
 
     pre_process_options(argc, argv);
 
@@ -720,7 +759,7 @@
 }
 
 #if !MICROPY_VFS
-uint mp_import_stat(const char *path) {
+mp_import_stat_t mp_import_stat(const char *path) {
     struct stat st;
     if (stat(path, &st) == 0) {
         if (S_ISDIR(st.st_mode)) {
