--- lsb_vsx-2.0-1/test_sets/SRC/install/scripts/config.sh	2022-07-18 12:39:27.958039930 +0200
+++ lsb_vsx-2.0-1-host/test_sets/SRC/install/scripts/config.sh	2022-09-08 18:07:53.300761413 +0200
@@ -110,49 +110,15 @@
 	fi
 }
 
-# Ask some Questions
-echo "
-In each of the following questions, the default answer in square
-brackets will be used if a carriage return alone is typed.
-If
-	!shell command
-
-is typed, the command will be sent to sh(1) to be executed.
-\`!' on its own stands for \`!sh'.
-"
-
-echo "*** The VSX source directory is rooted at $VSXDIR
-"
-
 TGTDIR="$VSXDIR"
 
-while :
-do
-	echo "** List of Available Parameter Files **
-"
-	ls -C params.data
-	getans "
-Which of the above parameter files do you wish to use
-in this installation procedure [none]? "
-	if test -z "$ans"
+	if test -f "$TGTDIR/vsxparams"
 	then
-		if test -f "$TGTDIR/vsxparams"
-		then
-			PARAMS="$TGTDIR/vsxparams"
-		else
-			PARAMS=params.data/.defaults
-		fi
-		break
+		PARAMS="$TGTDIR/vsxparams"
 	else
-		if test ! -f "params.data/$ans"
-		then
-			echo "*** Could not find $ans"
-		else
-			PARAMS=params.data/$ans
-			break
-		fi
+		PARAMS=params.data/.defaults
 	fi
-done
+
 SAVEVSXD="$VSXDIR"
 SAVEPATH="$PATH"
 echo "Setting defaults from: $PARAMS"
@@ -290,22 +256,9 @@
 	done
 	;;
 esac
-while :
-do
-	getans "
-Which test mode do you require
-(from:$modelist) [$default]? "
-	if test -z "$ans"
-	then
-		ans=$default
-	fi
-	case " $modelist " in
-	*" $ans "*)
-		TEST_MODE="$ans"
-		break
-		;;
-	esac
-done
+
+#Setting test mode
+TEST_MODE="POSIX96"
 
 # Set DEFINES based on test mode
 
@@ -438,14 +391,6 @@
 else
 	default=$VSX_OPER
 fi
-getans "
-What is your name [$default]? "
-if test -z "$ans"
-then
-	VSX_OPER="$default"
-else
-	VSX_OPER="$ans"
-fi
 
 if test -z "$VSX_ORG"
 then
@@ -453,14 +398,6 @@
 else
 	default=$VSX_ORG
 fi
-getans "
-What is the agency for whom you are running VSX [$default]? "
-if test -z "$ans"
-then
-	VSX_ORG="$default"
-else
-	VSX_ORG="$ans"
-fi
 
 if test -z "$VSX_SYS"
 then
@@ -468,14 +405,7 @@
 else
 	default=$VSX_SYS
 fi
-getans "
-What is the system on which you are running VSX [$default]? "
-if test -z "$ans"
-then
-	VSX_SYS="$default"
-else
-	VSX_SYS="$ans"
-fi
+
 
 if test -z "$TET_EXECUTE"
 then
@@ -483,147 +413,38 @@
 else
 	default=$TET_EXECUTE
 fi
-getans "
-Where is the root directory of the tree from which the
-testsets will be executed [$default]? "
-if test -z "$ans"
+echo $TET_EXECUTE
+if test ! -d "$TET_EXECUTE"
 then
-	ans=$default
+	echo "*** warning: directory $TET_EXECUTE does not exist"
 fi
-if test ! -d "$ans"
-then
-	echo "*** warning: directory $ans does not exist"
-	echo "*** it will be created by the VSX install procedure"
-fi
-TET_EXECUTE="$ans"
 
+# Keep default speed
+
+SPEED="5"
 if test -z "$SPEED"
 then
 	default="5"
 else
 	default=$SPEED
 fi
-while :
-do
-	getans "
-How fast is your machine (1=very fast, 10=slow) [$default]? "
-	if test -z "$ans"
-	then
-		ans="$default"
-	fi
-	case "$ans" in
-	1|2|3|4|5|6|7|8|9|10)
-		SPEED="$ans"
-		break;
-		;;
-	*)
-		echo "*** $ans is not a valid speed"
-		;;
-	esac
-done
 
+INCDIRS="/usr/include /usr/include/x86_64-linux-gnu"
 if test -z "$INCDIRS"
 then
-	INCDIRS="/usr/include"
+	INCDIRS="/usr/include /usr/include/x86_64-linux-gnu"
 fi
-while :
-do
-	getans "
-The system include directories (in order of searching) are:
-	$INCDIRS
-Is this correct (y/n) [y]? "
-	case "$ans" in
-	"y"|"")
-		confirm="1"
-		break
-		;;
-	"n")
-		confirm=""
-		break
-		;;
-	*)
-		;;
-	esac
-done
 
-if test -z "$confirm"
-then
-	INCDIRS=""
-	echo "
-Name the system include directories, in order of searching:"
-	while :
-	do
-		getans "	Name of next directory (type return to end): "
-		if test -z "$ans"
-		then
-			if test -z "$INCDIRS"
-			then
-				echo "*** Must name at least one directory"
-			else
-				break
-			fi
-		fi
-		if test ! -d "$ans"
-		then
-			echo "*** Could not find the directory $ans"
-		else
-			INCDIRS="$INCDIRS $ans"
-		fi
-	done
-fi
-
-
-if test -z "$CC"
-then
-	default=`mywhere c89`
-	if test -z "$default"
-	then
-		default=`mywhere cc`
-	fi
-	if test -z "$default"
-	then
-		default="c89"
-		# if selected, will be reported not found
-	fi
-else
-	default=$CC
-fi
-warntxt=""
-case $TEST_MODE in
-UNIX98)
-	warntxt=" (must be c89 for
-UNIX98 registration runs)"
-	;;
-esac
-while :
-do
-	getans "
-What is your C compiler called$warntxt [$default]? "
-	if test -z "$ans"
-	then
-		ans="$default"
-	fi
-	if test -z "`mywhere $ans`"
-	then
-		echo "*** Could not find $ans"
-	else
-		CC="`mywhere "$ans"`"
-		break
-	fi
-done
+CC=gcc
+default=$CC
 
+COPTS=""
 if test -z "$COPTS"
 then
 	default="none"
 else
 	default=$COPTS
 fi
-getans "
-What command line options does your C compiler need [$default]? "
-if test ! -z "$ans" 
-then
-	COPTS="$ans"
-fi
 
 if test -z "$LDFLAGS"
 then
@@ -631,12 +452,6 @@
 else
 	default=$LDFLAGS
 fi
-getans "
-What link editor options does your C compiler need [$default]? "
-if test ! -z "$ans" 
-then
-	LDFLAGS="$ans"
-fi
 
 echo "
 Searching for archive maintenance tools ..."
@@ -815,75 +630,6 @@
 	;;
 esac
 
-if test -z "$SYSLIBS"
-then
-	question="
-There are no additional libraries needed to compile VSX
-Is this correct (y/n) [y]? "
-else
-	question="
-The following additional libraries are needed to compile VSX
-	$SYSLIBS
-Is this correct (y/n) [y]? "
-fi
-while :
-do
-	getans "$question"
-	case "$ans" in
-	"y"|"")
-		confirm="1"
-		break
-		;;
-	"n")
-		confirm=""
-		break
-		;;
-	*)
-		;;
-	esac
-done
-
-if test -z "$confirm"
-then
-	LIBSTD=""
-	echo "
-Name the additional libraries needed, in order of searching
-(should be of form: -l<name> or full pathname): "
-	while :
-	do
-		getans "	Name of next library (type return to end): "
-		case $ans in
-		"")
-			break
-			;;
-		-l*)
-			# no quotes on $ans to allow for multiple libs
-			# that need each other to link OK
-			if checklib $ans
-			then
-				LIBSTD="$LIBSTD $ans"
-			else
-				echo "*** Could not find library $ans"
-			fi
-			;;
-		/*)
-			if test -f "$ans"
-			then
-				LIBSTD="$LIBSTD $ans"
-			else
-				echo "*** Could not find library $ans"
-			fi
-			;;
-		*)
-			echo "*** $ans not of form: -l<name> or full pathname"
-			;;
-		esac
-	done
-	SYSLIBS="$LIBSTD"
-else
-	LIBSTD="$SYSLIBS"
-fi
-
 # Only ask for XNFS_CLIENT, XNFS_SERVER and SERVER_TET_ROOT if testing XNFS
 
 case "$TEST_XNFS" in
@@ -957,35 +703,12 @@
 esac
 
 # Only ask for VPRINTF_SUPP if installing in a POSIX mode
+# Phoenix-RTOS support vprintf
 
 case "$TEST_MODE" in
 POSIX*|FIPS*)
-	if test -z "$VPRINTF_SUPP"
-	then
-		default=y
-	else
-		default=$VPRINTF_SUPP
-	fi
-	while :
-	do
-		getans "
-Is the ANSI function vprintf() supported (y/n) [$default]? "
-		if test -z "$ans"
-		then
-			ans=$default
-		fi
-		case "$ans" in
-		y|n)
-			VPRINTF_SUPP=$ans
-			break
-			;;
-		*)
-			;;
-		esac
-	done
-	;;
-*)
 	VPRINTF_SUPP=y
+	default=y
 	;;
 esac
 
@@ -1088,19 +811,6 @@
 		default=$CFPURE
 		textplus="(type \"none\" if no such option exists) "
 	fi
-	getans "
-Which command line option does your C compiler use to create a pure
-executable (shared text) image $textplus[$default]? "
-	case "$ans" in
-	"")
-		;;
-	*none*)
-		CFPURE=""
-		;;
-	*)
-		CFPURE="$ans"
-		;;
-	esac
 	;;
 esac
 
@@ -1118,62 +828,12 @@
 	else
 		default="$MLIB"
 	fi
-	while :
-	do
-		getans "
-Which library is your maths library [$default]?
-(should be of form: -l<name> or full pathname): "
-		case $ans in
-		"")
-			break
-			;;
-		-l*)
-			if checklib "$ans"
-			then
-				MLIB="$ans"
-				break
-			else
-				echo "*** Could not find library $ans"
-			fi
-			;;
-		/*)
-			if test -f "$ans"
-			then
-				MLIB="$ans"
-				break
-			else
-				echo "*** Could not find library $ans"
-			fi
-			;;
-		*)
-			echo "*** $ans not of form: -l<name> or full pathname"
-			;;
-		esac
-	done
 	;;
 esac
 
 case "$REALLY_ASK_NOSPC_DEV" in
 Y)
 	default="$NOSPC_DEV"
-	while :
-	do
-		getans "
-What is the mountable device to be used for ENOSPC testing [$default]? "
-		case $ans in
-		"")
-			ans="$default"
-			;;
-		esac
-		case $ans in
-		"")
-			;;
-		*)
-			NOSPC_DEV="$ans"
-			break
-			;;
-		esac
-	done
 	;;
 esac
 
@@ -1340,7 +1000,8 @@
 	done
 
 	(
-		sed "s/^all:.*/all:	privchk mktroot $dirgid_target install chmogpriv $filldisc_target $user_targets $fixed_targets/" \
+	# We build only this targets
+		sed "s/^all:.*/all:     install $dirgid_target mktroot /" \
 			"$VSXDIR"/install/Makefile.top &&
 		cd $VSXDIR/install/targets &&
 		cat $dirgid_target $filldisc_target $user_targets \
		
