--- lsb_vsx-2.0-1/test_sets/SRC/install/scripts/install.sh	2022-07-22 10:57:39.567599982 +0200
+++ lsb_vsx-2.0-1-phoenix/test_sets/SRC/install/scripts/install.sh	2022-09-22 11:43:09.630952562 +0200
@@ -11,16 +11,20 @@
 #		Outputs - Installation Report				#
 #########################################################################
 
+#
+#delete 2 function routines checks because they are compiled using phoenix-gcc
+#thus cannot be run on host
+#
+
 trap "" 1 2 3
 
 tfile=/tmp/ch.ss.tmp		# miscellaneous temporary files
 cfile=/tmp/ch.ss.c
 cdirfile=/tmp/chd.ss.c
 tdir=/tmp/ch.dr
-ifile=/tmp/ch.ss.i
 
 #  Remove temporary files and exit if a hangup or interrupt occurs
-rm_tmp1="rm -f /tmp/ans a.out $tfile $cfile"
+rm_tmp1="rm -f $tfile $cfile"
 rm_tmp2="rm -f $cdirfile $ifile"
 trap "$rm_tmp1; $rm_tmp2; exit 1" 1 2
 
@@ -134,6 +138,10 @@
     -e "/LONG_DOUBLE_SUPPORTED/s/ORTED.*$/ORTED	$DEF_LONG_DOUBLE/" \
 	../INC/std.h.orig > ../INC/std.h
 
+# This define resolves obsolete I_PUSH from ioctl.h
+
+echo "#define I_PUSH 0x5420" >> ../INC/std.h
+
 #########################################################################
 #									#
 # Create the include files for use by VSX				#
@@ -157,152 +165,6 @@
 
 #########################################################################
 #									#
-# Check whether the directory search routines work			#
-#									#
-#########################################################################
-
-cat > $cdirfile << 'E_O_F'
-#include <stdio.h>
-#include <sys/types.h>
-#include <dirent.h>
-main()
-{
-	DIR	*d;
-	struct	dirent	*dp;
-	char	*i;
-	int	c=0;
-
-	d = opendir(".");
-	while ((dp = readdir(d)) != 0)
-	{
-		i = (char *) dp->d_name;
-		c++;
-	}
-	closedir(d);
-	printf("%d\n", c);
-}
-E_O_F
-
-$CC -I../INC -I../SYSINC $DEFINES $COPTS "$cdirfile" $SYSLIBS > $tfile 2>&1
-
-if test "$?" != "0"
-then
-	echo "
-*** Could not compile test program for directory routines:
-
-$CC -I../INC -I../SYSINC $DEFINES $COPTS $cdirfile $SYSLIBS"
-	cat $tfile
-	echo "
-Check SRC/SYSINC/dirent.h before re-running install.sh"
-	exit 1
-fi
-
-mkdir $tdir
-touch $tdir/a $tdir/asjd $tdir/jsdhjas
-touch $tdir/kajshdjas $tdir/JH
-dir=`pwd`
-export dir
-(cd $tdir ; $dir/a.out > /tmp/ans 2>&1)
-rm -rf $tdir
-x=`cat /tmp/ans`
-if test "$x" != 7 -a "$x" != 5
-then
-	echo "
-*** Directory routines don't work!
-Check SRC/SYSINC/dirent.h before re-running install.sh"
-	exit 1
-fi
-
-#########################################################################
-#									#
-# Check whether variable argument lists work				#
-#									#
-#########################################################################
-
-cat << 'E_O_F' >$cfile
-#include <std.h>
-#include <stdlib.h>
-#if TEST_ANSI
-#include <stdarg.h>
-static int sub(char *, ...);
-#else
-#include <varargs.h>
-static int sub();
-#endif
-
-static char	arg1[] = "abc";
-static int	arg2 = 12345;
-static double	arg3 = 123.456;
-static char	arg4[] = "xyz";
-
-main(c, v)
-int c;
-char *v[];
-{
-	exit(sub(arg1, arg2, arg3, arg4));
-}
-
-static int
-#if TEST_ANSI
-sub(char *sarg1, ...)
-#else
-sub(sarg1, va_alist)
-char *sarg1;
-va_dcl
-#endif
-{
-	va_list ap;
-
-	if (sarg1 != arg1)
-		return 1;
-	
-#if TEST_ANSI
-	va_start(ap, sarg1);
-#else
-	va_start(ap);
-#endif
-
-	if (va_arg(ap, int) != arg2)
-		return 2;
-	
-	if (va_arg(ap, double) != arg3)
-		return 3;
-	
-	if (va_arg(ap, char *) != arg4)
-		return 4;
-	
-	va_end(ap);
-
-	return 0;
-}
-E_O_F
-
-$CC -I../INC -I../SYSINC $DEFINES $COPTS "$cfile" $SYSLIBS > $tfile 2>&1
-
-if test "$?" != "0" || grep arg "$tfile" > /dev/null 2>&1
-then
-	echo "
-*** Could not compile test program for variable argument lists
-
-$CC -I../INC -I../SYSINC $DEFINES $COPTS $cfile $SYSLIBS"
-	cat $tfile
-	echo "
-Check <stdarg.h>/<varargs.h> before re-running install.sh"
-	exit 1
-fi
-
-./a.out
-x=$?
-if test "$x" -ne 0
-then
-	echo "
-*** Variable argument lists don't work!  (test program exit code: $x)
-Check <stdarg.h>/<varargs.h> before re-running install.sh"
-	exit 1
-fi
-
-#########################################################################
-#									#
 # Set up the testroot 							#
 #									#
 #########################################################################
