diff -Naur lsb_vsx-2.0-1/src/defines.mk lsb_vsx-2.0-1-host/src/defines.mk
--- lsb_vsx-2.0-1/src/defines.mk	2022-07-18 12:39:27.910040986 +0200
+++ lsb_vsx-2.0-1-host/src/defines.mk	2022-08-03 17:05:08.933685765 +0200
@@ -153,7 +153,7 @@
 # Set SHLIB_COPTS to SHLIB_NOT_SUPPORTED if shared libraries are not supported
 # on your system or if you don't want to build the shared versions of the
 # API libraries.
-SHLIB_COPTS = -fPIC
+SHLIB_COPTS = SHLIB_NOT_SUPPORTED
 #
 # SHLIB_CC names the compiler that is used when generating the object files
 # that are to be put in a shared library.
@@ -179,7 +179,7 @@
 #
 # C_PLUS - the name of the C++ compiler
 # To disable C++ support, set C_PLUS = CPLUSPLUS_NOT_SUPPORTED.
-C_PLUS = g++
+C_PLUS = CPLUSPLUS_NOT_SUPPORTED
 # C_SUFFIX - suffix for C++ source files
 # (without an initial dot; e.g: C, cpp etc.)
 C_SUFFIX = C
diff -Naur lsb_vsx-2.0-1/test_sets/scripts/common/psl_setup.sh lsb_vsx-2.0-1-host/test_sets/scripts/common/psl_setup.sh
--- lsb_vsx-2.0-1/test_sets/scripts/common/psl_setup.sh	2022-07-18 12:39:27.962039842 +0200
+++ lsb_vsx-2.0-1-host/test_sets/scripts/common/psl_setup.sh	2022-08-08 15:46:47.327527660 +0200
@@ -27,12 +27,12 @@
 then
   touch scripts/locks/psl_setup
 
-  printf "Install pseudolanguages ..? [y] "
-  read cmd
-  if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
-          PSLDEFS_DIR="psldefs"
-	  printf "Enter the root password:\n"
-	  su root -c "cd $TOOLDIR/$PSLDEFS_DIR && make"
-  fi
+#  printf "Install pseudolanguages ..? [y] "
+#  read cmd
+#  if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
+#          PSLDEFS_DIR="psldefs"
+#	  printf "Enter the root password:\n"
+#	  su root -c "cd $TOOLDIR/$PSLDEFS_DIR && make"
+#  fi
 fi
   
diff -Naur lsb_vsx-2.0-1/test_sets/scripts/vsx-pcts/post-build.sh lsb_vsx-2.0-1-host/test_sets/scripts/vsx-pcts/post-build.sh
--- lsb_vsx-2.0-1/test_sets/scripts/vsx-pcts/post-build.sh	2022-07-18 12:39:27.962039842 +0200
+++ lsb_vsx-2.0-1-host/test_sets/scripts/vsx-pcts/post-build.sh	2022-08-08 17:00:52.001383703 +0200
@@ -1,7 +1,8 @@
 #!/bin/sh
 
-printf "Install scripts into TESTROOT/BIN..? [y] "
-read cmd
+#printf "Install scripts into TESTROOT/BIN..? [y] "
+#read cmd
+cmd=y
 if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
         echo
 	echo "The test suites requires the generation of certain files."
@@ -10,8 +11,9 @@
 	echo "by default on your system. If you do not use pax you can expect"
 	echo "to see more test failures."
 	echo 
-	printf "Use pax..? [y] "
-	read cmd
+	#printf "Use pax..? [y] "
+	#read cmd
+	cmd=y
 	if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
 	    CPIO_SCRIPT_TOKEN="#PAX"
 	    TAR_SCRIPT_TOKEN="#PAX"
@@ -41,9 +43,9 @@
 #
 #
 
-printf "Install nonexistent devices ..? [y] "
-read cmd
-if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
-	printf "Enter the root password:\n"
-	su root -c "sh scripts/vsx-pcts/nonexist_device.setup"
-fi
+#printf "Install nonexistent devices ..? [y] "
+#read cmd
+#if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
+#	printf "Enter the root password:\n"
+#	su root -c "sh scripts/vsx-pcts/nonexist_device.setup"
+#fi
diff -Naur lsb_vsx-2.0-1/test_sets/SRC/common/vprog/vmake.sh lsb_vsx-2.0-1-host/test_sets/SRC/common/vprog/vmake.sh
--- lsb_vsx-2.0-1/test_sets/SRC/common/vprog/vmake.sh	2022-07-18 12:39:27.954040018 +0200
+++ lsb_vsx-2.0-1-host/test_sets/SRC/common/vprog/vmake.sh	2022-08-23 18:18:22.976834133 +0200
@@ -256,6 +256,10 @@
 	;;
 esac
 
+#move executable to TESTROOT
+mv "$TET_EXECUTE/../tset/$sect/$area/$tset"/T.$i "$Mtdir"/T.$i
+exit 0
+
 if test "$macros" = n
 then
 	# Create a dummy macro testset
diff -Naur lsb_vsx-2.0-1/test_sets/SRC/install/Makefile.mid lsb_vsx-2.0-1-host/test_sets/SRC/install/Makefile.mid
--- lsb_vsx-2.0-1/test_sets/SRC/install/Makefile.mid	2022-07-18 12:39:27.958039930 +0200
+++ lsb_vsx-2.0-1-host/test_sets/SRC/install/Makefile.mid	2022-08-08 16:18:55.075164768 +0200
@@ -19,8 +19,6 @@
 	test -d $(TET_EXECUTE) || { \
 	    mkdir $(TET_EXECUTE) && \
 	    touch $(TET_EXECUTE)/.keepme && \
-	    $(CHOWN) vsx0 $(TET_EXECUTE) && \
-	    $(CHGRP) vsxg0 $(TET_EXECUTE) && \
 	    $(CHMOD) u=rwx,g=rx,o=rx $(TET_EXECUTE); \
 	}
 
diff -Naur lsb_vsx-2.0-1/test_sets/SRC/install/Makefile.nfs lsb_vsx-2.0-1-host/test_sets/SRC/install/Makefile.nfs
--- lsb_vsx-2.0-1/test_sets/SRC/install/Makefile.nfs	2022-07-18 12:39:27.958039930 +0200
+++ lsb_vsx-2.0-1-host/test_sets/SRC/install/Makefile.nfs	2022-08-08 16:20:39.673442927 +0200
@@ -48,7 +48,7 @@
 # Run install.sh as user vsx0, ensuring HOME is set to vsx0 home directory
 
 install:
-	su vsx0 -c 'HOME=`pwd` BIN/install.sh'
+	su 'whoami' -c 'HOME=`pwd` BIN/install.sh'
 
 # Assign privileges to the chmog program
 
diff -Naur lsb_vsx-2.0-1/test_sets/SRC/install/Makefile.top lsb_vsx-2.0-1-host/test_sets/SRC/install/Makefile.top
--- lsb_vsx-2.0-1/test_sets/SRC/install/Makefile.top	2022-07-18 12:39:27.958039930 +0200
+++ lsb_vsx-2.0-1-host/test_sets/SRC/install/Makefile.top	2022-08-08 16:21:14.078200464 +0200
@@ -35,19 +35,22 @@
 # Check make is running with the necessary privileges
 
 privchk:
-	@if test -z "`id | grep 'uid=0[^0-9]' || true`"; then \
-	    echo >&2 "make must be run as super-user"; \
-	    exit 1; \
-	fi
+	#we don't want to use root
+	#@if test -z "`id | grep 'uid=0[^0-9]' || true`"; then \
+	    #echo >&2 "make must be run as super-user"; \
+	    #exit 1; \
+	#fi
 
 # Run install.sh as user vsx0, ensuring HOME is set to vsx0 home directory
 
 install:
-	su vsx0 -c 'HOME=`pwd` BIN/install.sh'
+	#run as owner
+	HOME=`pwd` BIN/install.sh
 
 # Assign privileges to the chmog program
 
 chmogpriv:	$(CHMOG)
-	chown root $(CHMOG)
-	chmod u+s $(CHMOG)
+	#we don't want to use root
+	#chown root $(CHMOG)
+	#chmod u+s $(CHMOG)
 
diff -Naur lsb_vsx-2.0-1/test_sets/SRC/install/targets/filldisc lsb_vsx-2.0-1-host/test_sets/SRC/install/targets/filldisc
--- lsb_vsx-2.0-1/test_sets/SRC/install/targets/filldisc	2022-07-18 12:39:27.962039842 +0200
+++ lsb_vsx-2.0-1-host/test_sets/SRC/install/targets/filldisc	2022-08-08 16:04:25.003303100 +0200
@@ -3,11 +3,11 @@
 # Initialise file system for ENOSPC tests
 
 MOUNT_DIR = /tmp/vsxmnt
-
+#don't mount loopback disk for now
 filldisc:
-	test -d $(MOUNT_DIR) || mkdir $(MOUNT_DIR)
-	mount $(NOSPC_DEV) $(MOUNT_DIR)
-	cd $(MOUNT_DIR) && $(SCRIPTS)/filldisc.sh
-	sync; sleep 2; df $(MOUNT_DIR)
-	-umount $(MOUNT_DIR)
+	#test -d $(MOUNT_DIR) || mkdir $(MOUNT_DIR)
+	#mount $(NOSPC_DEV) $(MOUNT_DIR)
+	#cd $(MOUNT_DIR) && $(SCRIPTS)/filldisc.sh
+	#sync; sleep 2; df $(MOUNT_DIR)
+	#-umount $(MOUNT_DIR)
 
--- lsb_vsx-2.0-1/test_sets/SRC/subsets/base/config_sub.sh	2022-07-18 12:39:27.962039842 +0200
+++ lsb_vsx-2.0-1-host/test_sets/SRC/subsets/base/config_sub.sh	2022-09-08 12:40:38.763291114 +0200
@@ -4,36 +4,15 @@
 # Ask for SIGNAL_SUPP if installing in a POSIX mode
 # and not testing XNFS
 
+
+#Phoenix-RTOS support signal function
+
 case "$TEST_MODE:$TEST_XNFS" in
 *:[yY])
 	;;
 POSIX*|FIPS*)
-	if test -z "$SIGNAL_SUPP"
-	then
-		default=y
-	else
-		default=$SIGNAL_SUPP
-	fi
-	while :
-	do
-		getans "
-Is the ANSI function signal() supported (y/n) [$default]? "
-		if test -z "$ans"
-		then
-			ans=$default
-		fi
-		case "$ans" in
-		y|n)
-			SIGNAL_SUPP=$ans
-			break
-			;;
-		*)
-			;;
-		esac
-	done
-	;;
-*)
 	SIGNAL_SUPP=y
+	default=y
 	;;
 esac
 
--- lsb_vsx-2.0-1/test_sets/SRC/common/vport/nonposix.c	2022-07-18 12:39:27.954040018 +0200
+++ lsb_vsx-2.0-1-host/test_sets/SRC/common/vport/nonposix.c	2022-09-08 12:43:36.264008695 +0200
@@ -282,75 +282,7 @@
 
 #endif /* __STDC__ */
 
-#ifndef __linux__
-
-public	int	opterr = 1;
-public	int	optind = 1;
-public	char	*optarg;
-
-public int
-getopt(argc, argv, opts)
-int	argc;
-char	**argv, *opts;
-{
-	static int sp = 1;
-	int c;
-	char *cp;
-
-	if (sp == 1)
-	{
-		if (optind >= argc ||
-		   argv[optind][0] != '-' || argv[optind][1] == '\0')
-			return EOF;
-		else if(strcmp(argv[optind], "--") == 0)
-		{
-			optind++;
-			return EOF;
-		}
-	}
-
-	c = argv[optind][sp];
-	if (c == ':' || (cp=strchr(opts, c)) == NULL)
-	{
-		if (opterr)
-			(void) fprintf(stderr, "%s: illegal option -- %c\n",
-				argv[0], c);
-		if (argv[optind][++sp] == '\0')
-		{
-			optind++;
-			sp = 1;
-		}
-		return '?';
-	}
-	if (*++cp == ':')
-	{
-		if (argv[optind][sp+1] != '\0')
-			optarg = &argv[optind++][sp+1];
-		else if(++optind >= argc)
-		{
-			if (opterr)
-				(void) fprintf(stderr,
-				    "%s: option requires an argument -- %c\n",
-				    argv[0], c);
-			sp = 1;
-			return '?';
-		} else
-			optarg = argv[optind++];
-		sp = 1;
-	}
-	else
-	{
-		if (argv[optind][++sp] == '\0')
-		{
-			sp = 1;
-			optind++;
-		}
-		optarg = NULL;
-	}
-	return c;
-}
-
-#endif /* __linux__ */
+/* Phoenix lib contains getopt func */
 
 public int
 isnan(x)
@@ -359,123 +291,8 @@
 	return (x != x);
 }
 
-#ifndef __linux__
-
-#define	tst(a, b) (*mode == 'r' ? (b) : (a))
-
-static pid_t *popen_pid = NULL;
-
-public FILE *
-popen(cmd, mode)
-char	*cmd, *mode;
-{
-	int	p[2];
-	int	myside, yourside;
-	pid_t	*poptr, pid;
-
-	static long open_max;
-
-	if (popen_pid == NULL)
-	{
-		errno = 0;
-		open_max = sysconf(_SC_OPEN_MAX);
-		if (open_max < 0 && errno == 0)
-			open_max = 256;
-		if (open_max < 0 ||
-		   (popen_pid = (pid_t *)calloc((size_t) open_max,
-		   sizeof(pid_t))) == NULL)
-			return NULL;
-	}
-
-	if (pipe(p) < 0)
-		return NULL;
-
-	myside = tst(p[1], p[0]);
-	yourside = tst(p[0], p[1]);
-
-	if ((pid = fork()) == 0)
-	{
-		/* myside and yourside reverse roles in child */
-		int	stdio;
-
-		/* close all pipes from other popen's */
-		poptr = &popen_pid[open_max];
-		while (poptr > popen_pid)
-		{
-			if (*--poptr != 0)
-				(void) close(poptr - popen_pid);
-		}
-		stdio = tst(0, 1);
-		(void) close(myside);
-		(void) close(stdio);
-		(void) fcntl(yourside, F_DUPFD, stdio);
-		(void) close(yourside);
-		(void) execlp("sh", "sh", "-c", cmd, (char *)0);
-		_exit(1);
-	}
-	else if (pid == -1)
-		return NULL;
-
-	popen_pid[myside] = pid;
-	(void) close(yourside);
-
-	return fdopen(myside, mode);
-}
-
-public int
-pclose(ptr)
-FILE	*ptr;
-{
-	int f, status, err = errno;
-	pid_t p;
-	sigset_t set, oset;
-
-	if (popen_pid == NULL)
-	{
-		/* no popen() calls made */
-		errno = ECHILD;
-		return SYSERROR;
-	}
-
-	f = fileno(ptr);
-	p = popen_pid[f];
-	(void) fclose(ptr);
-	popen_pid[f] = 0;
-
-	if (p <= 0)
-	{
-		/* no child to wait for */
-		errno = ECHILD;
-		return SYSERROR;
-	}
-
-	(void) sigemptyset(&set);
-	(void) sigaddset(&set, SIGHUP);
-	(void) sigaddset(&set, SIGINT);
-	(void) sigaddset(&set, SIGQUIT);
-	(void) sigprocmask(SIG_BLOCK, &set, &oset);
-
-	if (waitpid(p, &status, 0) != p)
-	{
-		status = SYSERROR;
-		err = errno;
-	}
-
-	(void) sigprocmask(SIG_SETMASK, &oset, (sigset_t *)NULL);
-	if (sigpending(&set) != SYSERROR &&
-	   (sigismember(&set, SIGHUP) ||
-	    sigismember(&set, SIGINT) ||
-	    sigismember(&set, SIGQUIT)))
-	{
-		(void) sigsuspend(&oset);
-		err = errno;
-	}
-
-	errno = err;
-	return status;
-}
 
-#endif /* __linux__ */
+/*Phoenix lib contains popen and pclose func */
 
 #if !(__STDC__ > 0) && !ANSI_VPRINTF_FUNC 
 
--- lsb_vsx-2.0-1/src/tet3/dtet2lib/notty.c	2022-07-18 12:39:27.906041074 +0200
+++ lsb_vsx-2.0-1-host/src/tet3/dtet2lib/notty.c	2022-09-08 13:56:32.235616417 +0200
@@ -47,7 +47,8 @@
 #    include <unistd.h>
 #    include <errno.h>
 #  else /* HAS_SETSID */
-#    include <sys/ioctl.h>
+/*Change obsolete name, headers are the same */
+#    include <sys/stropts.h>
 #    ifdef TIOCNOTTY
 #      include <fcntl.h>
 #    endif /* TIOCNOTTY */
--- lsb_vsx-2.0-1/LSB.tools/userintf.c	2022-07-18 12:39:27.902041162 +0200
+++ lsb_vsx-2.0-1-host/LSB.tools/userintf.c	2022-09-08 14:09:19.353973329 +0200
@@ -589,7 +589,7 @@
 /* #define STREAMS_PTY	/* uncomment if pty's use STREAMS */
 
 #ifdef STREAMS_PTY
-#include <sys/stropts.h>
+#include <sys/ioctl.h>
 #endif
 
 public int
--- lsb_vsx-2.0-1/test_sets/SRC/install/userintf/gen_termios	2022-07-18 12:39:27.962039842 +0200
+++ lsb_vsx-2.0-1-host/test_sets/SRC/install/userintf/gen_termios	2022-09-08 14:10:42.963443090 +0200
@@ -69,7 +69,7 @@
 /* #define STREAMS_PTY	/* uncomment if pty's use STREAMS */
 
 #ifdef STREAMS_PTY
-#include <sys/stropts.h>
+#include <sys/ioctl.h>
 #endif
 
 public int
--- lsb_vsx-2.0-1/test_sets/SRC/common/vtools/hdrallow.c	2022-07-18 12:39:27.954040018 +0200
+++ lsb_vsx-2.0-1-host/test_sets/SRC/common/vtools/hdrallow.c	2022-09-08 14:11:33.396393202 +0200
@@ -719,7 +719,7 @@
 			return 1;
 #endif
 	}
-	else if (strcmp(hdrname, "stropts.h") == 0)
+	else if (strcmp(hdrname, "ioctl.h") == 0)
 	{
 		/* Allow "bi_.*", "ic_.*", "l_.*", "sl_.*" and "str_.*" */
 
--- lsb_vsx-2.0-1/test_sets/SRC/common/vport/userintfES.c	2022-07-18 12:39:27.954040018 +0200
+++ lsb_vsx-2.0-1-host/test_sets/SRC/common/vport/userintfES.c	2022-09-08 14:12:30.233514551 +0200
@@ -727,7 +727,7 @@
 #define STREAMS_PTY	/* uncomment if pty's use STREAMS */
 
 #ifdef STREAMS_PTY
-#include <sys/stropts.h>
+#include <sys/ioctl.h>
 #endif
 
 public int

--- lsb_vsx-2.0-1/test_sets/SRC/install/scripts/install.sh	2022-07-22 10:57:39.567599982 +0200
+++ lsb_vsx-2.0-1-host/test_sets/SRC/install/scripts/install.sh	2022-09-08 15:41:05.054383634 +0200
@@ -134,6 +134,10 @@
     -e "/LONG_DOUBLE_SUPPORTED/s/ORTED.*$/ORTED	$DEF_LONG_DOUBLE/" \
 	../INC/std.h.orig > ../INC/std.h
 
+# This define resolves obsolete I_PUSH from ioctl.h
+
+echo "#define I_PUSH TIOCPKT" >> ../INC/std.h
+
 #########################################################################
 #									#
 # Create the include files for use by VSX				#
--- lsb_vsx-2.0-1/test_sets/setup_testsets.sh	2022-07-18 12:39:27.950040106 +0200
+++ lsb_vsx-2.0-1-host/test_sets/setup_testsets.sh	2022-09-08 18:15:32.848014622 +0200
@@ -135,16 +135,17 @@
 
 #---------------------------------------------------------------------
 # Pre Configure testset scripts
-run_testset_config_script pre-config
+#don' t create loopback disk for now
+run_testset_config_script #pre-config
 
 ######################################################################
 # Run config script
-printf "Run config.sh ..? [y] "
-read cmd
-if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
+#printf "Run config.sh ..? [y] "
+#read cmd
+#if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
 	config.sh
-fi
-printf "\n----------------------------------------------------------\n"
+#fi
+#printf "\n----------------------------------------------------------\n"
 
 # Common cleanup needed
 # Fixup NSIG to _NSIG which common to all Linux's (and CLK_TCK)
@@ -163,10 +164,10 @@
 ######################################################################
 # Build the test framework
 cd $INSTDIR
-
 printf "\n\n----------------------------------------------------------\n"
 printf "Building the test framework ... enter the root password\n"
-su root -c make
+#we don't want to use root
+make
 if [ $? -ne 0 ]; then
   echo "Test framework compilation failed. Aborting"
   exit 1
@@ -186,8 +187,9 @@
 
 ######################################################################
 # Parameterisation
-printf "Install tetexec.cfg ..? [y] "
-read cmd
+#printf "Install tetexec.cfg ..? [y] "
+#read cmd
+cmd=y
 if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
 
   echo "Parameterizing the $TET_EXECUTE/tetexec.cfg file"
@@ -215,13 +217,15 @@
 
 fi
 
+# We only want to build test environment tools needed for building tests on host
+
 ######################################################################
 # Build & Execute testsets
-printf "Build and Execute testsets ..? [y]"
-read cmd
-if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
-  sh run_testsets.sh
-fi
+#printf "Build and Execute testsets ..? [y]"
+#read cmd
+#if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
+#  sh run_testsets.sh
+#fi
 
 echo 
 echo "In order to re run the test suites at a later date run the"
 
