--- lsb_vsx_cp2/test_sets/run_testsets.sh	2023-03-07 11:54:08.364161127 +0100
+++ lsb_vsx_cp/test_sets/run_testsets.sh	2023-06-27 17:29:49.849537003 +0200
@@ -8,25 +8,17 @@
 printf "\n\n----------------------------------------------------------------------\n"
 printf "Building and Executing testsets\n"
 
-printf "Build testsets ..? [y]"
-read cmd
-if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
-  tcc -p -b -s $HOME/scen.bld $*
-fi
-
-printf "\n\n----------------------------------------------------------------------\n"
-printf "Executing testsets\n"
-tcc -p -e -s $HOME/scen.exec $*
-(cd results; awk -f `which vres` `ls -d 0* | tail -1`/journal)
-printf "A formal report has been saved in $HOME/results/report\n"
-(cd results; vrpt `ls -d 0* | tail -1`/journal >$HOME/results/report)
-
-printf "Rerun just the failures (produces a journal of only fails) ..? [y] "
-read cmd
-if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
-  JNL=`ls -d $HOME/results/0* | tail -1`
-  JNLDIR=`basename $JNL`
-  echo "running tcc -p -e -s scen.exec -r FAIL,UNRESOLVED,UNINITIATED results/$JNLDIR/journal"
-  tcc -p -e -s scen.exec -r FAIL,UNRESOLVED,UNINITIATED results/$JNLDIR/journal
-fi
-
+#copy bin and test_sets/BIN folders with programs compiled under phoenix to temporary folder
+#these are needed to exec tcc for executing tests
+mv -f ../bin/* ../../tmp/tetbin_phoenix
+mv -f ../test_sets/BIN/* ../../tmp/BIN_phoenix
+#copy bin and test_sets/BIN folders compiled under host
+cp ../../tmp/tetbin_host/* ../bin
+cp ../../tmp/BIN_host/* ../test_sets/BIN
+#execute build
+tcc -p -b -s ${PREFIX_LSB_VSX_CONFIG}/files/scen.bld -n .hdr
+#clean unnecessary after build files
+rm -f ../bin/* ../test_sets/BIN/*
+#bring back phoenix bin and test_sets/BIN for forward test execution
+cp ../../tmp/tetbin_phoenix/* ../bin
+cp ../../tmp/BIN_phoenix/* ../test_sets/BIN
