diff -Naur lsb_vsx-2.0-1/src/tet3/tcclib/tcfexec.c lsb_vsx-2.0-1-phoenix/src/tet3/tcclib/tcfexec.c
--- lsb_vsx-2.0-1/src/tet3/tcclib/tcfexec.c	2022-07-18 12:39:27.910040986 +0200
+++ lsb_vsx-2.0-1-phoenix/src/tet3/tcclib/tcfexec.c	2022-08-23 11:05:08.099451404 +0200
@@ -165,7 +165,8 @@
 				_exit(~0 - 1);
 			}
 		}
-		for (n = tet_getdtablesize() - 1; n > 2; n--) {
+		/*Value set manually cause tet_getdtablesize returns int max value */
+		for (n = 511; n > 2; n--) {
 			if (n != fd)
 				(void) close(n);
 		}
