--- lsb_vsx-2.0-1/test_sets/setup_testsets.sh	2022-07-18 12:39:27.950040106 +0200
+++ lsb_vsx-2.0-1-phoenix/test_sets/setup_testsets.sh	2022-09-22 12:39:24.169812675 +0200
@@ -135,15 +135,15 @@
 
 #---------------------------------------------------------------------
 # Pre Configure testset scripts
-run_testset_config_script pre-config
+#don' t create loopback disk for now
+run_testset_config_script #pre-config
 
 ######################################################################
 # Run config script
-printf "Run config.sh ..? [y] "
-read cmd
-if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
-	config.sh
-fi
+# Copy vsxparams.mk to params.data
+cp $PREFIX_LSB_VSX/lsb_vsx-2.0-1-config/files/vsxparams.mk $INSTDIR/SRC/install/params.data
+echo "Running config.sh"
+echo "vsxparams.mk\n" | config.sh
 printf "\n----------------------------------------------------------\n"
 
 # Common cleanup needed
@@ -163,10 +163,10 @@
 ######################################################################
 # Build the test framework
 cd $INSTDIR
-
 printf "\n\n----------------------------------------------------------\n"
-printf "Building the test framework ... enter the root password\n"
-su root -c make
+#Deletion of prompting for ansewers, answers set manually
+#we don't want to use root
+make
 if [ $? -ne 0 ]; then
   echo "Test framework compilation failed. Aborting"
   exit 1
@@ -186,8 +186,9 @@
 
 ######################################################################
 # Parameterisation
-printf "Install tetexec.cfg ..? [y] "
-read cmd
+#printf "Install tetexec.cfg ..? [y] "
+#read cmd
+cmd=y
 if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
 
   echo "Parameterizing the $TET_EXECUTE/tetexec.cfg file"
@@ -217,13 +218,7 @@
 
 ######################################################################
 # Build & Execute testsets
-printf "Build and Execute testsets ..? [y]"
-read cmd
-if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
-  sh run_testsets.sh
+if [ "$PLATFORM" = "PHOENIX-RTOS" ]; then
+	echo "Building and Executing testsets"
+  	sh run_testsets.sh
 fi
-
-echo 
-echo "In order to re run the test suites at a later date run the"
-echo "rerun_tests program in vsx0's home directory as the vsx0 user"
-echo
