--- lsb_vsx-2.0-1/test_sets/SRC/install/targets/filldisc	2022-07-18 12:39:27.962039842 +0200
+++ lsb_vsx-2.0-1-phoenix/test_sets/SRC/install/targets/filldisc	2022-08-25 12:48:42.224018354 +0200
@@ -3,11 +3,11 @@
 # Initialise file system for ENOSPC tests
 
 MOUNT_DIR = /tmp/vsxmnt
-
+#don't mount loopback disk for now!
 filldisc:
-	test -d $(MOUNT_DIR) || mkdir $(MOUNT_DIR)
-	mount $(NOSPC_DEV) $(MOUNT_DIR)
-	cd $(MOUNT_DIR) && $(SCRIPTS)/filldisc.sh
-	sync; sleep 2; df $(MOUNT_DIR)
-	-umount $(MOUNT_DIR)
+	#test -d $(MOUNT_DIR) || mkdir $(MOUNT_DIR)
+	#mount $(NOSPC_DEV) $(MOUNT_DIR)
+	#cd $(MOUNT_DIR) && $(SCRIPTS)/filldisc.sh
+	#sync; sleep 2; df $(MOUNT_DIR)
+	#-umount $(MOUNT_DIR)
 
diff -Naur lsb_vsx-2.0-1/test_sets/scripts/common/psl_setup.sh lsb_vsx-2.0-1-phoenix/test_sets/scripts/common/psl_setup.sh
--- lsb_vsx-2.0-1/test_sets/scripts/common/psl_setup.sh	2022-07-18 12:39:27.962039842 +0200
+++ lsb_vsx-2.0-1-phoenix/test_sets/scripts/common/psl_setup.sh	2022-08-23 12:10:08.423705255 +0200
@@ -27,12 +27,14 @@
 then
   touch scripts/locks/psl_setup
 
-  printf "Install pseudolanguages ..? [y] "
-  read cmd
-  if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
-          PSLDEFS_DIR="psldefs"
-	  printf "Enter the root password:\n"
-	  su root -c "cd $TOOLDIR/$PSLDEFS_DIR && make"
-  fi
+#Psuedolanguages not needed for now
+
+#  printf "Install pseudolanguages ..? [y] "
+#  read cmd
+#  if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
+#          PSLDEFS_DIR="psldefs"
+#	  printf "Enter the root password:\n"
+#	  su root -c "cd $TOOLDIR/$PSLDEFS_DIR && make"
+#  fi
 fi
   
diff -Naur lsb_vsx-2.0-1/test_sets/scripts/vsx-pcts/post-build.sh lsb_vsx-2.0-1-phoenix/test_sets/scripts/vsx-pcts/post-build.sh
--- lsb_vsx-2.0-1/test_sets/scripts/vsx-pcts/post-build.sh	2022-07-18 12:39:27.962039842 +0200
+++ lsb_vsx-2.0-1-phoenix/test_sets/scripts/vsx-pcts/post-build.sh	2022-08-23 12:23:45.822080962 +0200
@@ -1,7 +1,10 @@
 #!/bin/sh
 
-printf "Install scripts into TESTROOT/BIN..? [y] "
-read cmd
+#Deletion of prompting for answer, answers set manually
+
+#printf "Install scripts into TESTROOT/BIN..? [y] "
+#read cmd
+cmd=y
 if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
         echo
 	echo "The test suites requires the generation of certain files."
@@ -10,8 +13,9 @@
 	echo "by default on your system. If you do not use pax you can expect"
 	echo "to see more test failures."
 	echo 
-	printf "Use pax..? [y] "
-	read cmd
+	#printf "Use pax..? [y] "
+	#read cmd
+	cmd=y
 	if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
 	    CPIO_SCRIPT_TOKEN="#PAX"
 	    TAR_SCRIPT_TOKEN="#PAX"
@@ -37,13 +41,15 @@
 #	fi
 fi
 
+#Don't install nonexisiting devices, we don't use root
+
 # Installation of nonexistent devices
 #
 #
 
-printf "Install nonexistent devices ..? [y] "
-read cmd
-if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
-	printf "Enter the root password:\n"
-	su root -c "sh scripts/vsx-pcts/nonexist_device.setup"
-fi
+#printf "Install nonexistent devices ..? [y] "
+#read cmd
+#if [ "$cmd" = "Y" -o "$cmd" = "y" -o "$cmd" = "" ] ; then
+#	printf "Enter the root password:\n"
+#	su root -c "sh scripts/vsx-pcts/nonexist_device.setup"
+#fi

diff -Naur lsb_vsx-2.0-1/test_sets/SRC/common/genlib/Makefile.org lsb_vsx-2.0-1-phoenix/test_sets/SRC/common/genlib/Makefile.org
--- lsb_vsx-2.0-1/test_sets/SRC/common/genlib/Makefile.org	2022-07-18 12:39:27.950040106 +0200
+++ lsb_vsx-2.0-1-phoenix/test_sets/SRC/common/genlib/Makefile.org	2022-08-23 17:35:43.000869951 +0200
@@ -19,7 +19,7 @@
 #	TARGETS:	../../LIB/genlib.a
 ##########################################################################
 
-CC =	/bin/cc
+CC =	 ~/toolchains/$(HOST)/$(HOST)/bin/$(CC)
 AR =	ar cr
 RM =		rm -f
 LINT =	lint
@@ -50,8 +50,8 @@
 INSTINC =
 
 INSTLINT =	$(LINTDIR)/llib-lgen.ln
-INSTOWN =	vsx0
-INSTGRP =	vsxg0
+INSTOWN =	$(USER)
+INSTGRP =	$(USER)
 INSTMODE =	644
 
 LOCALINC =
@@ -67,11 +67,12 @@
 CFLOCAL =
 CFLAGS =	$(CFLOCAL) $(INCS) $(DEFINES) $(DBUG) $(COPTS) $(SDBFLAG)
 
+		#do not use locals for now
 CFILES =	alrm.c asc_to_psl.c atouid.c childsync.c chktimes.c \
 		clean_dir.c closesync.c cppair.c mcppair.c crdir.c \
 		crfile.c eaccess.c errname.c execpath.c hexstr.c \
 		itoa.c killchild.c ltoa.c fpcomp.c opensync.c parsync.c \
-		pr_char.c pr_wchar.c pslanginfo.c pslclass.c pslocale.c \
+		pr_char.c pr_wchar.c pslanginfo.c pslclass.c \
 		psl_to_asc.c  sccsid.c sendsync.c \
 		uids.c uidtoa.c vxmkdir.c waitsync.c xx_all.c
 
@@ -79,7 +80,7 @@
 		clean_dir.o closesync.o cppair.o mcppair.o crdir.o \
 		crfile.o eaccess.o errname.o execpath.o hexstr.o \
 		itoa.o killchild.o ltoa.o fpcomp.o opensync.o parsync.o \
-		pr_char.o pr_wchar.o pslanginfo.o pslclass.o pslocale.o \
+		pr_char.o pr_wchar.o pslanginfo.o pslclass.o \
 		psl_to_asc.o  sccsid.o sendsync.o \
 		uids.o uidtoa.o vxmkdir.o waitsync.o xx_all.o
 --- lsb_vsx-2.0-1/test_sets/SRC/install/targets/filldisc	2022-07-18 12:39:27.962039842 +0200
+++ lsb_vsx-2.0-1-phoenix/test_sets/SRC/install/targets/filldisc	2022-08-25 12:48:42.224018354 +0200
@@ -3,11 +3,11 @@
 # Initialise file system for ENOSPC tests
 
 MOUNT_DIR = /tmp/vsxmnt
-
+#don't mount loopback disk for now!
 filldisc:
-	test -d $(MOUNT_DIR) || mkdir $(MOUNT_DIR)
-	mount $(NOSPC_DEV) $(MOUNT_DIR)
-	cd $(MOUNT_DIR) && $(SCRIPTS)/filldisc.sh
-	sync; sleep 2; df $(MOUNT_DIR)
-	-umount $(MOUNT_DIR)
+	#test -d $(MOUNT_DIR) || mkdir $(MOUNT_DIR)
+	#mount $(NOSPC_DEV) $(MOUNT_DIR)
+	#cd $(MOUNT_DIR) && $(SCRIPTS)/filldisc.sh
+	#sync; sleep 2; df $(MOUNT_DIR)
+	#-umount $(MOUNT_DIR)
 
