diff -Naur lsb_vsx-2.0-1/test_sets/SRC/common/wchars/wcc.c lsb_vsx-2.0-1-phoenix/test_sets/SRC/common/wchars/wcc.c
--- lsb_vsx-2.0-1/test_sets/SRC/common/wchars/wcc.c	2022-07-18 12:39:27.954040018 +0200
+++ lsb_vsx-2.0-1-phoenix/test_sets/SRC/common/wchars/wcc.c	2022-07-29 13:15:51.692511217 +0200
@@ -94,6 +94,10 @@
 #define ASSIGN_CHAR	'='
 #define PROPERTY	9999
 
+/* posix test */
+#define MB_CUR_MAX 4
+/* posix test */
+
 #define CONVERT_CHECK(_addr, _size1, _size2, _size3)				\
 		if (str1 != NULL)				\
 			str1 = convert_field(str1, _size1, _size2, _size3);		

diff -Naur lsb_vsx-2.0-1/test_sets/SRC/common/vlib/sigvars.c lsb_vsx-2.0-1-phoenix/test_sets/SRC/common/vlib/sigvars.c
--- lsb_vsx-2.0-1/test_sets/SRC/common/vlib/sigvars.c	2022-07-18 12:39:27.954040018 +0200
+++ lsb_vsx-2.0-1-phoenix/test_sets/SRC/common/vlib/sigvars.c	2022-08-23 16:31:28.701336673 +0200
@@ -56,6 +56,9 @@
 #include	<vsx_signal.h>
 #include	<sysdep.h>
 
+/*Added this define for compatibility */
+#define SIGPOLL SIGIO
+
 public	int	sig_array[VNUMSIG + 1] = {
 	0,
 	SIGABRT,
diff -Naur lsb_vsx-2.0-1/test_sets/SRC/common/vtools/obstack.h lsb_vsx-2.0-1-phoenix/test_sets/SRC/common/vtools/obstack.h
--- lsb_vsx-2.0-1/test_sets/SRC/common/vtools/obstack.h	2022-07-18 12:39:27.954040018 +0200
+++ lsb_vsx-2.0-1-phoenix/test_sets/SRC/common/vtools/obstack.h	2022-07-29 10:41:16.985076180 +0200
@@ -287,7 +287,9 @@
    : (int) obst_free ((h), (h)->temp + (char *) (h)->chunk)))
 #endif
 
+/* declared in libphoenix/include/string.h 
 extern void bcopy();
+*/
 
 #endif /* not __OBSTACKS__ */
 
