diff -Naur lsb_vsx-2.0-1/test_sets/SRC/common/wchars/Makefile.org lsb_vsx-2.0-1-phoenix/test_sets/SRC/common/wchars/Makefile.org
--- lsb_vsx-2.0-1/test_sets/SRC/common/wchars/Makefile.org	2022-07-18 12:39:27.954040018 +0200
+++ lsb_vsx-2.0-1-phoenix/test_sets/SRC/common/wchars/Makefile.org	2022-08-23 17:14:36.613944437 +0200
@@ -71,10 +71,11 @@
 all:		$(INSTDIR)/wchars.bin
 
 $(INSTDIR)/wchars.bin:	wchars.bin
-		cp $? $@
-		-$(CHOWN) $(INSTOWN) $@
-		-$(CHGRP) $(INSTGRP) $@
-		-$(CHMOD) $(INSTMODE) $@
+		#disable wchars for now
+		#cp $? $@
+		#-$(CHOWN) $(INSTOWN) $@
+		#-$(CHGRP) $(INSTGRP) $@
+		#-$(CHMOD) $(INSTMODE) $@
 
 wchars.bin:	wcc $(VSXDIR)/wchars.cfg
 		./wcc $(VSXDIR)/wchars.cfg $@
		
