--- lsb_vsx-2.0-1/test_sets/SRC/common/genlib/Makefile.org	2022-07-18 12:39:27.950040106 +0200
+++ lsb_vsx-2.0-1-phoenix/test_sets/SRC/common/genlib/Makefile.org	2022-09-09 10:21:17.927287680 +0200
@@ -19,7 +19,7 @@
 #	TARGETS:	../../LIB/genlib.a
 ##########################################################################
 
-CC =	/bin/cc
+CC =	$(BUF_CC)
 AR =	ar cr
 RM =		rm -f
 LINT =	lint
@@ -50,8 +50,8 @@
 INSTINC =
 
 INSTLINT =	$(LINTDIR)/llib-lgen.ln
-INSTOWN =	vsx0
-INSTGRP =	vsxg0
+INSTOWN =	$(USER)
+INSTGRP =	$(USER)
 INSTMODE =	644
 
 LOCALINC =
@@ -67,11 +67,12 @@
 CFLOCAL =
 CFLAGS =	$(CFLOCAL) $(INCS) $(DEFINES) $(DBUG) $(COPTS) $(SDBFLAG)
 
+		#do not use locals for now
 CFILES =	alrm.c asc_to_psl.c atouid.c childsync.c chktimes.c \
 		clean_dir.c closesync.c cppair.c mcppair.c crdir.c \
 		crfile.c eaccess.c errname.c execpath.c hexstr.c \
 		itoa.c killchild.c ltoa.c fpcomp.c opensync.c parsync.c \
-		pr_char.c pr_wchar.c pslanginfo.c pslclass.c pslocale.c \
+		pr_char.c pr_wchar.c pslanginfo.c pslclass.c \
 		psl_to_asc.c  sccsid.c sendsync.c \
 		uids.c uidtoa.c vxmkdir.c waitsync.c xx_all.c
 
@@ -79,7 +80,7 @@
 		clean_dir.o closesync.o cppair.o mcppair.o crdir.o \
 		crfile.o eaccess.o errname.o execpath.o hexstr.o \
 		itoa.o killchild.o ltoa.o fpcomp.o opensync.o parsync.o \
-		pr_char.o pr_wchar.o pslanginfo.o pslclass.o pslocale.o \
+		pr_char.o pr_wchar.o pslanginfo.o pslclass.o \
 		psl_to_asc.o  sccsid.o sendsync.o \
 		uids.o uidtoa.o vxmkdir.o waitsync.o xx_all.o



