--- lsb_vsx/tset/POSIX.os/sysdb/getpwnam/getpwnam.c	2024-10-16 15:22:06.485884602 +0200
+++ lsb_vsx.phoenix/tset/POSIX.os/sysdb/getpwnam/getpwnam.c	2024-10-16 15:22:06.485884602 +0200
@@ -122,8 +122,9 @@
 
 	DBUG_ENTER("test1");
 
+	/* There is only 1 user - root */
 	errno = 0;
-	if((rval = tb_getpwuid((uid_t) vguid0)) == (struct passwd *) NULL)
+	if((rval = tb_getpwuid((uid_t) 0)) == (struct passwd *) NULL)
 	{
 		err = errno;
 		xx_rpt(DELETION);
@@ -133,10 +134,10 @@
 	else
 		PATH_TRACE;
 
-	if(rval->pw_uid != (uid_t) vguid0)
+	if(rval->pw_uid != (uid_t) 0)
 	{
 		xx_rpt(DELETION);
-		in_rpt("getpwuid has returned the incorrect user - expected %d, got %d",(uid_t)vguid0,rval->pw_uid);
+		in_rpt("getpwuid has returned the incorrect user - expected %d, got %d",(uid_t)0,rval->pw_uid);
 		in_rpt("name of user found %s",rval->pw_name);
 		DBUG_VOID_RETURN;
 	}
@@ -144,7 +145,7 @@
 		PATH_TRACE;
 
 	errno = 0;
-	if ((name = newstr(rval->pw_name)) == NULL)
+	if ((name = newstr("root")) == NULL)
 	{
 		err = errno;
 		xx_rpt(DELETION);
--- lsb_vsx/tset/POSIX.os/sysdb/getpwuid/getpwuid.c	2024-10-16 15:22:06.485884602 +0200
+++ lsb_vsx.phoenix/tset/POSIX.os/sysdb/getpwuid/getpwuid.c	2024-10-16 15:22:06.485884602 +0200
@@ -119,7 +119,8 @@
 	DBUG_ENTER("test1");
 
 	errno = 0;
-	if((rval = tb_getpwuid((uid_t) vguid0)) == (struct passwd *) NULL)
+	/* There is only 1 user - root */
+	if((rval = tb_getpwuid((uid_t) 0)) == (struct passwd *) NULL)
 	{
 		err = errno;
 		xx_rpt(FAILURE);
@@ -129,11 +130,11 @@
 	else
 		PATH_TRACE;
 
-	if(rval->pw_uid != (uid_t) vguid0)
+	if(rval->pw_uid != (uid_t) 0)
 	{
 		xx_rpt(FAILURE);
 		in_rpt("getpwuid has returned the incorrect user - expected %d, got %ld",
-			vguid0, (long)rval->pw_uid);
+			0, (long)rval->pw_uid);
 		in_rpt("name of user found %s",rval->pw_name);
 		DBUG_VOID_RETURN;
 	}
