diff --git a/cognit/include/logger.h b/cognit/include/logger.h
index 69fea0d..09f20ac 100644
--- a/cognit/include/logger.h
+++ b/cognit/include/logger.h
@@ -6,30 +6,40 @@
 #define COGNIT_LOG_LEVEL_INFO  2
 #define COGNIT_LOG_LEVEL_ERROR 3
 
-#define COGNIT_LOG_LEVEL COGNIT_LOG_LEVEL_DEBUG
+#ifndef COGNIT_LOG_LEVEL
+#define COGNIT_LOG_LEVEL COGNIT_LOG_LEVEL_INFO
+#endif
 
 #define COGNIT_LOG_TRACE(...)                       \
     if (COGNIT_LOG_LEVEL <= COGNIT_LOG_LEVEL_TRACE) \
+    {                                               \
         printf("[TRACE] ");                         \
-    printf(__VA_ARGS__);                            \
-    printf("\n");
+        printf(__VA_ARGS__);                        \
+        printf("\n");                               \
+    }
 
 #define COGNIT_LOG_DEBUG(...)                       \
     if (COGNIT_LOG_LEVEL <= COGNIT_LOG_LEVEL_DEBUG) \
+    {                                               \
         printf("[DEBUG] ");                         \
-    printf(__VA_ARGS__);                            \
-    printf("\n");
+        printf(__VA_ARGS__);                        \
+        printf("\n");                               \
+    }
 
 #define COGNIT_LOG_INFO(...)                       \
     if (COGNIT_LOG_LEVEL <= COGNIT_LOG_LEVEL_INFO) \
+    {                                              \
         printf("[INFO] ");                         \
-    printf(__VA_ARGS__);                           \
-    printf("\n");
+        printf(__VA_ARGS__);                       \
+        printf("\n");                              \
+    }
 
 #define COGNIT_LOG_ERROR(...)                       \
     if (COGNIT_LOG_LEVEL <= COGNIT_LOG_LEVEL_ERROR) \
+    {                                               \
         printf("[ERROR] ");                         \
-    printf(__VA_ARGS__);                            \
-    printf("\n");
+        printf(__VA_ARGS__);                        \
+        printf("\n");                               \
+    }
 
 #endif // LOGGER_HEADER_
