diff -ur '--exclude=*.git' mbedtls-2.28.0/programs/ssl/ssl_context_info.c mbedtls-2.28.0-phoenix/programs/ssl/ssl_context_info.c
--- mbedtls-2.28.0/programs/ssl/ssl_context_info.c	2022-05-12 15:49:04.327065471 +0200
+++ mbedtls-2.28.0-phoenix/programs/ssl/ssl_context_info.c	2022-05-12 15:44:21.216459932 +0200
@@ -118,7 +118,8 @@
 FILE *b64_file = NULL;                  /* file with base64 codes to deserialize */
 char conf_keep_peer_certificate = 1;    /* MBEDTLS_SSL_KEEP_PEER_CERTIFICATE from mbedTLS configuration */
 char conf_dtls_proto = 1;               /* MBEDTLS_SSL_PROTO_DTLS from mbedTLS configuration */
-char debug = 0;                         /* flag for debug messages */
+/* 'debug' has been already defined as syscall in Phoenix-RTOS */
+static char debug = 0;                  /* flag for debug messages */
 const char alloc_err[] = "Cannot allocate memory\n";
 const char buf_ln_err[] = "Buffer does not have enough data to complete the parsing\n";
