diff -ur '--exclude=*.git' mbedtls-2.28.0/Makefile mbedtls-2.28.0-phoenix/Makefile
--- mbedtls-2.28.0/Makefile	2022-05-12 15:49:04.287065127 +0200
+++ mbedtls-2.28.0-phoenix/Makefile	2022-05-12 15:40:36.482048098 +0200
@@ -1,4 +1,4 @@
-DESTDIR=/usr/local
+DESTDIR=$(PREFIX_BUILD)
 PREFIX=mbedtls_
 
 .SILENT:
diff -ur '--exclude=*.git' mbedtls-2.28.0/programs/Makefile mbedtls-2.28.0-phoenix/programs/Makefile
--- mbedtls-2.28.0/programs/Makefile	2022-05-12 15:49:04.323065436 +0200
+++ mbedtls-2.28.0-phoenix/programs/Makefile	2022-05-12 15:42:00.290993638 +0200
@@ -146,11 +146,13 @@
 
 all: $(APPS)
 ifndef WINDOWS
+ifndef phoenix
 # APPS doesn't include the fuzzing programs, which aren't "normal"
 # sample or test programs, and don't build with MSVC which is
 # warning about fopen
 all: fuzz
 endif
+endif
 
 fuzz: ${MBEDTLS_TEST_OBJS}
 	$(MAKE) -C fuzz THIRDPARTY_INCLUDES=$(THIRDPARTY_INCLUDES)
diff -ur '--exclude=*.git' mbedtls-2.28.0/programs/ssl/ssl_context_info.c mbedtls-2.28.0-phoenix/programs/ssl/ssl_context_info.c
--- mbedtls-2.28.0/programs/ssl/ssl_context_info.c	2022-05-12 15:49:04.327065471 +0200
+++ mbedtls-2.28.0-phoenix/programs/ssl/ssl_context_info.c	2022-05-12 15:44:21.216459932 +0200
@@ -118,7 +118,8 @@
 FILE *b64_file = NULL;                  /* file with base64 codes to deserialize */
 char conf_keep_peer_certificate = 1;    /* MBEDTLS_SSL_KEEP_PEER_CERTIFICATE from mbedTLS configuration */
 char conf_dtls_proto = 1;               /* MBEDTLS_SSL_PROTO_DTLS from mbedTLS configuration */
-char debug = 0;                         /* flag for debug messages */
+/* 'debug' has been already defined as global variable in Phoenix-RTOS */
+char mbedtls_debug = 0;                         /* flag for debug messages */
 const char alloc_err[] = "Cannot allocate memory\n";
 const char buf_ln_err[] = "Buffer does not have enough data to complete the parsing\n";
 
@@ -154,7 +155,7 @@
 
 void printf_dbg( const char *str, ... )
 {
-    if( debug )
+    if( mbedtls_debug )
     {
         va_list args;
         va_start( args, str );
@@ -206,7 +207,7 @@
     {
         if( strcmp( argv[i], "-d" ) == 0 )
         {
-            debug = 1;
+            mbedtls_debug = 1;
         }
         else if( strcmp( argv[i], "-h" ) == 0 )
         {
@@ -1087,7 +1088,7 @@
                 continue;
             }
 
-            if( debug )
+            if( mbedtls_debug )
             {
                 printf( "\nDecoded data in hex:\n\t");
                 print_hex( ssl_buf, ssl_len, 25, "\t" );
diff -ur '--exclude=*.git' mbedtls-2.28.0/programs/ssl/ssl_server2.c mbedtls-2.28.0-phoenix/programs/ssl/ssl_server2.c
--- mbedtls-2.28.0/programs/ssl/ssl_server2.c	2022-05-12 15:49:04.327065471 +0200
+++ mbedtls-2.28.0-phoenix/programs/ssl/ssl_server2.c	2022-05-12 15:47:42.042345370 +0200
@@ -1720,7 +1720,8 @@
 #if defined(_MSC_VER)
             opt.renego_period = _strtoui64( q, NULL, 10 );
 #else
-            if( sscanf( q, "%" SCNu64, &opt.renego_period ) != 1 )
+            /* no SCNu64 define in Phoenix-RTOS */
+            if( sscanf( q, "%u", &opt.renego_period ) != 1 )
                 goto usage;
 #endif /* _MSC_VER */
             if( opt.renego_period < 2 )