--- lsb_vsx/src/tet3/apilib/shared.mk	2024-10-16 15:22:06.485884602 +0200
+++ lsb_vsx.host/src/tet3/apilib/shared.mk	2024-10-16 15:22:06.485884602 +0200
@@ -179,6 +179,9 @@
 	$(LOCAL_CC) $(DTET_CFLAGS) -c $(APISRC)sigreset.c
 
 
+# Ports are build with -j 9, below targets cannot be run in parallel due to mv command
+.NOTPARALLEL: tet_1fork$O tet_afork$O
+
 # dependencies
 
 apichk.o: $(DINC)/tet_api.h $(INC)/apilib.h $(INC)/dtmac.h $(INC)/ltoa.h
