diff -ur '--exclude=*.git' azure-iot-sdk-c-lts_01_2022/CMakeLists.txt azure-iot-sdk-c-lts_01_2022-phoenix/CMakeLists.txt
--- azure-iot-sdk-c-lts_01_2022/CMakeLists.txt	2022-09-08 15:13:41.900770742 +0200
+++ azure-iot-sdk-c-lts_01_2022-phoenix/CMakeLists.txt	2022-09-08 15:11:36.382335799 +0200
@@ -83,6 +83,8 @@
 option(use_microsoftazure_de_cert "set use_microsoftazure_de_cert to ON if the MicrosoftAzure DE cert is to be used for samples, set to OFF to not use it" OFF)
 option(use_portal_azure_cn_cert "set use_portal_azure_cn_cert to ON if the Portal Azure CN cert is to be used for samples, set to OFF to not use it" OFF)
 
+set(warnings_as_errors OFF)
+
 ########## Conditional Options set ##########
 if (WIN32 OR MACOSX)
     option(use_openssl "set use_openssl to ON to use OpenSSL." OFF)
diff -ur '--exclude=*.git' azure-iot-sdk-c-lts_01_2022/c-utility/configs/azure_iot_build_rules.cmake azure-iot-sdk-c-lts_01_2022-phoenix/c-utility/configs/azure_iot_build_rules.cmake
--- azure-iot-sdk-c-lts_01_2022/c-utility/configs/azure_iot_build_rules.cmake	2022-09-08 15:01:17.386328626 +0200
+++ azure-iot-sdk-c-lts_01_2022-phoenix/c-utility/configs/azure_iot_build_rules.cmake	2022-09-08 15:14:57.154230605 +0200
@@ -71,8 +71,6 @@
       set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /W4")
       set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /W4")
 elseif(UNIX) #LINUX OR APPLE
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")
     if(NOT (IN_OPENWRT OR APPLE))
         set (CMAKE_C_FLAGS "-D_POSIX_C_SOURCE=200112L ${CMAKE_C_FLAGS}")
     endif()
@@ -209,10 +207,6 @@
     add_definitions(-D_CRT_SECURE_NO_WARNINGS)
     # Make warning as error
     add_definitions(/WX)
-ELSE()
-    # Make warning as error
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")
 ENDIF()
 
 
