diff -ur '--exclude=*.git' azure-iot-sdk-c-lts_01_2022/iothub_client/samples/iothub_ll_telemetry_sample/iothub_ll_telemetry_sample.c azure-iot-sdk-c-lts_01_2022-phoenix/iothub_client/samples/iothub_ll_telemetry_sample/iothub_ll_telemetry_sample.c
--- azure-iot-sdk-c-lts_01_2022/iothub_client/samples/iothub_ll_telemetry_sample/iothub_ll_telemetry_sample.c	2022-06-28 12:35:13.425760154 +0200
+++ azure-iot-sdk-c-lts_01_2022-phoenix/iothub_client/samples/iothub_ll_telemetry_sample/iothub_ll_telemetry_sample.c	2022-06-28 12:40:28.615709745 +0200
@@ -50,7 +50,7 @@
 
 
 /* Paste in the your iothub connection string  */
-static const char* connectionString = "[device connection string]";
+static const char* connectionString = "Your IoTHub Connection String";
 #define MESSAGE_COUNT        5
 static bool g_continueRunning = true;
 static size_t g_message_count_send_confirmations = 0;
