--- azure-iot-sdk-c/CMakeLists.txt	2022-03-21 13:46:15.658058907 +0100
+++ azure-iot-sdk-c-phoenix/CMakeLists.txt	2022-03-21 13:39:00.744676974 +0100
@@ -83,6 +83,8 @@
 option(use_microsoftazure_de_cert "set use_microsoftazure_de_cert to ON if the MicrosoftAzure DE cert is to be used for samples, set to OFF to not use it" OFF)
 option(use_portal_azure_cn_cert "set use_portal_azure_cn_cert to ON if the Portal Azure CN cert is to be used for samples, set to OFF to not use it" OFF)
 
+set(warnings_as_errors OFF)
+
 ########## Conditional Options set ##########
 if (WIN32 OR MACOSX)
     option(use_openssl "set use_openssl to ON to use OpenSSL." OFF)
