diff -ur '--exclude=*.git' azure-iot-sdk-c/iothub_client/samples/iothub_client_sample_mqtt_dm/CMakeLists.txt azure-iot-sdk-c-phoenix/iothub_client/samples/iothub_client_sample_mqtt_dm/CMakeLists.txt
--- azure-iot-sdk-c/iothub_client/samples/iothub_client_sample_mqtt_dm/CMakeLists.txt	2022-04-06 10:45:38.219356482 +0200
+++ azure-iot-sdk-c-phoenix/iothub_client/samples/iothub_client_sample_mqtt_dm/CMakeLists.txt	2022-04-06 10:48:48.898536998 +0200
@@ -39,13 +39,13 @@
 
     add_executable(iothub_client_sample_mqtt_dm ${iothub_client_sample_mqtt_dm_c_files} ${iothub_client_sample_mqtt_dm_h_files})
 
+    # pthread is present in libphoenix library, so there is no need to link it separately
     target_link_libraries(iothub_client_sample_mqtt_dm
         iothub_client
         iothub_client_mqtt_transport
         serializer
         aziotsharedutil
         umqtt
-        pthread
         curl
         m
     )
