# Fix: Make library installable and use stable, versioned dependencies.
diff -ruN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2025-08-19 09:57:32.296953644 +0200
+++ b/CMakeLists.txt	2025-09-09 10:26:06.053224322 +0200
@@ -13,23 +13,19 @@
 
 FetchContent_Declare(
   slice99
-  GIT_REPOSITORY https://github.com/Hirrolot/slice99
-  GIT_TAG 7075e1e)
+  URL https://github.com/hirrolot/slice99/archive/refs/tags/v0.7.8.tar.gz)
 
 FetchContent_Declare(
   metalang99
-  GIT_REPOSITORY https://github.com/Hirrolot/metalang99
-  GIT_TAG e0eb4e1)
+  URL https://github.com/hirrolot/metalang99/archive/refs/tags/v1.13.5.tar.gz)
 
 FetchContent_Declare(
   datatype99
-  GIT_REPOSITORY https://github.com/Hirrolot/datatype99
-  GIT_TAG aa24712)
+  URL https://github.com/hirrolot/datatype99/archive/refs/tags/v1.6.5.tar.gz)
 
 FetchContent_Declare(
   interface99
-  GIT_REPOSITORY https://github.com/Hirrolot/interface99
-  GIT_TAG 73c7c1e)
+  URL https://github.com/hirrolot/interface99/archive/refs/tags/v1.0.2.tar.gz)
 
 FetchContent_MakeAvailable(slice99 datatype99 interface99)
 
@@ -135,3 +131,25 @@
 target_link_libraries(${PROJECT_NAME} PUBLIC slice99 metalang99 datatype99 interface99)
 
 set_target_properties(${PROJECT_NAME} PROPERTIES C_STANDARD 99 C_STANDARD_REQUIRED ON)
+
+install(
+    TARGETS smolrtsp
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
+)
+
+install(
+    DIRECTORY ${PROJECT_SOURCE_DIR}/include/
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/include
+)
+
+FetchContent_GetProperties(slice99 SOURCE_DIR SLICE99_SOURCE_DIR)
+install(FILES ${SLICE99_SOURCE_DIR}/slice99.h DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
+
+FetchContent_GetProperties(metalang99 SOURCE_DIR METALANG99_SOURCE_DIR)
+install(DIRECTORY ${METALANG99_SOURCE_DIR}/include/ DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
+
+FetchContent_GetProperties(datatype99 SOURCE_DIR DATATYPE99_SOURCE_DIR)
+install(FILES ${DATATYPE99_SOURCE_DIR}/datatype99.h DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
+
+FetchContent_GetProperties(interface99 SOURCE_DIR INTERFACE99_SOURCE_DIR)
+install(FILES ${INTERFACE99_SOURCE_DIR}/interface99.h DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
