diff -ruN a/include/X11/Xalloca.h b/include/X11/Xalloca.h
--- a/include/X11/Xalloca.h	2025-02-19 14:55:32.463374051 +0100
+++ b/include/X11/Xalloca.h	2025-02-20 10:36:36.613372018 +0100
@@ -65,6 +65,7 @@
 #endif
 
 #ifdef INCLUDE_ALLOCA_H
+#  include <stddef.h>
 #  include <alloca.h>
 #endif
 
diff -ruN a/include/X11/Xos_r.h b/include/X11/Xos_r.h
--- a/include/X11/Xos_r.h	2025-02-19 14:58:24.895819569 +0100
+++ b/include/X11/Xos_r.h	2025-02-19 15:32:44.236582491 +0100
@@ -248,7 +248,7 @@
  */
 
 #if defined(__NetBSD__) || defined(__FreeBSD__) || defined(__OpenBSD__) || \
-    defined(__APPLE__) || defined(__DragonFly__)
+    defined(__APPLE__) || defined(__DragonFly__) || defined(__phoenix__)
 static __inline__ void _Xpw_copyPasswd(_Xgetpwparams p)
 {
    memcpy(&(p).pws, (p).pwp, sizeof(struct passwd));
@@ -261,11 +261,17 @@
    (p).len = strlen((p).pwp->pw_passwd);
    strcpy((p).pws.pw_passwd,(p).pwp->pw_passwd);
 
+#if !defined(__phoenix__)
    (p).pws.pw_class = (p).pws.pw_passwd + (p).len + 1;
    (p).len = strlen((p).pwp->pw_class);
    strcpy((p).pws.pw_class, (p).pwp->pw_class);
+#endif
 
+#if defined(__phoenix__)
+   (p).pws.pw_gecos = (p).pws.pw_passwd + (p).len + 1;
+#else
    (p).pws.pw_gecos = (p).pws.pw_class + (p).len + 1;
+#endif
    (p).len = strlen((p).pwp->pw_gecos);
    strcpy((p).pws.pw_gecos, (p).pwp->pw_gecos);
 
@@ -318,7 +324,7 @@
   (_Xos_processUnlock), \
   (p).pwp )
 
-#elif !defined(_POSIX_THREAD_SAFE_FUNCTIONS) && !defined(__APPLE__)
+#elif !defined(_POSIX_THREAD_SAFE_FUNCTIONS) && !defined(__APPLE__) && !defined(__phoenix__)
 # define X_NEEDS_PWPARAMS
 typedef struct {
   struct passwd pws;
@@ -407,6 +413,10 @@
 #include <sys/param.h>
 #endif
 
+#ifdef __phoenix__
+#define MAXHOSTNAMELEN HOST_NAME_MAX
+#endif
+
 typedef struct {
   struct hostent hent;
   char           h_name[MAXHOSTNAMELEN];
diff -ruN a/include/X11/Xpoll.h.in b/include/X11/Xpoll.h.in
--- a/include/X11/Xpoll.h.in	2025-02-19 14:55:32.463374051 +0100
+++ b/include/X11/Xpoll.h.in	2025-02-19 15:39:53.852526126 +0100
@@ -50,6 +50,10 @@
 #ifndef _XPOLL_H_
 #define _XPOLL_H_
 
+#ifdef __phoenix__
+typedef uint32_t fd_mask;
+#endif
+
 #if !defined(WIN32) || defined(__CYGWIN__)
 
 #ifndef USE_POLL
@@ -163,7 +167,11 @@
         }
 
 #else /* USE_POLL */
+#ifdef __phoenix__
+#include <poll.h>
+#else
 #include <sys/poll.h>
+#endif
 #endif /* USE_POLL */
 
 #else /* WIN32 */
