--- lsb_vsx/test_sets/SRC/subsets/base/config_sub.sh	2024-10-16 15:22:06.485884602 +0200
+++ lsb_vsx.phoenix/test_sets/SRC/subsets/base/config_sub.sh	2024-10-16 15:22:06.485884602 +0200
@@ -4,39 +4,6 @@
 # Ask for SIGNAL_SUPP if installing in a POSIX mode
 # and not testing XNFS
 
-case "$TEST_MODE:$TEST_XNFS" in
-*:[yY])
-	;;
-POSIX*|FIPS*)
-	if test -z "$SIGNAL_SUPP"
-	then
-		default=y
-	else
-		default=$SIGNAL_SUPP
-	fi
-	while :
-	do
-		getans "
-Is the ANSI function signal() supported (y/n) [$default]? "
-		if test -z "$ans"
-		then
-			ans=$default
-		fi
-		case "$ans" in
-		y|n)
-			SIGNAL_SUPP=$ans
-			break
-			;;
-		*)
-			;;
-		esac
-	done
-	;;
-*)
-	SIGNAL_SUPP=y
-	;;
-esac
-
 # set ASK_* variables for optional config parameters
 
 case $TEST_XNFS in
