--- lsb_vsx/test_sets/scripts/vsx-pcts/parameterisations.sh	2024-10-16 15:22:06.485884602 +0200
+++ lsb_vsx.phoenix/test_sets/scripts/vsx-pcts/parameterisations.sh	2024-10-16 15:22:06.485884602 +0200
@@ -63,24 +63,3 @@
 echo "VSX_LINE_BUF_SUPP=T" >> $1
 echo "VSX_NXIO_BLKDEV=$HOME/nonexistb" >>$1
 echo "VSX_NXIO_CHRDEV=$HOME/nonexistc" >>$1
-
-grep vsx1 /etc/passwd 2>&1 >/dev/null
-if [ $? -eq 1 ]
-then
-  printf "Warning: it appears as if the uids and gids are not correctly installed proceeding anyway..."
-else
-  VSX_UID0=`grep vsx0 /etc/passwd|cut -d: -f3`
-  VSX_GID0=`grep vsx0 /etc/passwd|cut -d: -f4`
-  VSX_UID1=`grep vsx1 /etc/passwd|cut -d: -f3`
-  VSX_GID1=`grep vsx1 /etc/passwd|cut -d: -f4`
-  VSX_UID2=`grep vsx2 /etc/passwd|cut -d: -f3`
-  VSX_GID2=`grep vsx2 /etc/passwd|cut -d: -f4`
-  
-  printf "VSX_UID0=$VSX_UID0\n" >> $1
-  printf "VSX_UID1=$VSX_UID1\n" >> $1
-  printf "VSX_UID2=$VSX_UID2\n" >> $1
-  printf "VSX_GID0=$VSX_GID0\n" >> $1
-  printf "VSX_GID1=$VSX_GID1\n" >> $1
-  printf "VSX_GID2=$VSX_GID2\n" >> $1
-fi
-
