--- lsb_vsx/test_sets/tset/ANSI.os/streamio/ungetc/ungetc.c	2024-10-16 15:22:06.485884602 +0200
+++ lsb_vsx.phoenix/test_sets/tset/ANSI.os/streamio/ungetc/ungetc.c	2024-10-16 15:22:06.485884602 +0200
@@ -172,6 +172,22 @@
 	(void) unlink(t_file);
 }
 
+/* gets() is a stub, add local definition */
+private char*
+local_gets(buffer)
+char *buffer;
+{
+	int ch;
+	char* p = buffer;
+
+	while ((ch = getchar()) != '\n' && ch != EOF) {
+		*p++ = ch;
+	}
+	*p = '\0';
+
+	return buffer;
+}
+
 
 private void
 test1()
@@ -257,7 +273,7 @@
 
 	if (ungetc(T6_CHAR, fp) != T6_CHAR)
 		fail = 1;
-	(void) gets(buf);
+	(void) local_gets(buf);
 	val6 = buf[0];
 	/* don't fclose(fp) */
 
