--- lsb_vsx/test_sets/tset/POSIX.os/procprim/kill/kill.c	2025-05-08 13:50:19.000000000 +0200
+++ lsb_vsx.phoenix/test_sets/tset/POSIX.os/procprim/kill/kill.c	2025-10-20 18:38:13.984483792 +0200
@@ -1137,8 +1137,8 @@
 		if (tm_out)
 		{
 			in_rpt("signal did not kill child");
-			for (i = 0 ; i < NCHILDREN ; i++)
-			DBUG_PRINTF("ch_t9and10", "kill_child %ld", (long)children[i]);
+			for (i = 0 ; i < NCHILDREN ; i++) {
+				DBUG_PRINTF("ch_t9and10", "kill_child %ld", (long)children[i]);
 				if ((rval = killchild(children[i],
 						WAITTIME)) != SUCCEED)
 				{
@@ -1146,6 +1146,7 @@
 						rval);
 					DBUG_EXIT(A_KILLFAIL);
 				}
+			}
 		}
 	}
 	else
