diff -u b/kdrive/phoenix/keyboard.c b/kdrive/phoenix/keyboard.c
--- b/kdrive/phoenix/keyboard.c	2024-08-28 18:27:18.002303324 +0200
+++ b/kdrive/phoenix/keyboard.c	2024-09-02 10:32:16.393897703 +0200
@@ -11,7 +11,8 @@
 
 static int PhoenixKbdFd;
 
-static const KeySym phoenix_to_x[128] = {
+static const KeySym phoenix_to_x[256] = {
+/* -- Normal -- */
 /* 0 unused */     NoSymbol,
 /* 1 ESCape */     XK_Escape,
 /* 2 1 */          XK_1,
@@ -54,7 +55,7 @@
 /* 39 ; */         XK_semicolon,
 /* 40 ' */         XK_apostrophe,
 /* 41 ` */         XK_grave,
-/* 42 shift */     XK_Shift_R,
+/* 42 shift */     XK_Shift_L,
 /* 43 \ */         XK_backslash,
 /* 44 z */         XK_z,
 /* 45 x */         XK_x,
@@ -66,7 +67,7 @@
 /* 51 , */         XK_comma,
 /* 52 . */         XK_period,
 /* 53 / */         XK_slash,
-/* 54 shift */     XK_Shift_L,
+/* 54 shift */     XK_Shift_R,
 /* 55 kp * */      NoSymbol, // TODO ?????
 /* 56 alt */       XK_Alt_L,
 /* 57 space */     XK_space,
@@ -103,8 +104,137 @@
 /* 88 f12 */       XK_F12,
 /* 89 0 */         NoSymbol,
 /* 90 0 */         NoSymbol,
-/* 91 0 */         NoSymbol,
-/* 92 0 */         NoSymbol,
+/* 91 0 */         XK_Super_L,
+/* 92 0 */         XK_Super_R,
+/* 93 0 */         NoSymbol,
+/* 94 0 */         NoSymbol,
+/* 95 0 */         NoSymbol,
+/* 96 extended */  NoSymbol,
+/* 97 0 */         NoSymbol,
+/* 98 0 */         NoSymbol,
+/* 99 0 */         NoSymbol,
+/* 100 */          NoSymbol,
+/* 101 */          NoSymbol,
+/* 102 */          NoSymbol,
+/* 103 */          XK_Up,
+/* 104 */          NoSymbol,
+/* 105 */          XK_Left,
+/* 106 */          XK_Right,
+/* 107 */          NoSymbol,
+/* 108 */          XK_Down,
+/* 109 */          NoSymbol,
+/* 110 */          NoSymbol,
+/* 111 */          NoSymbol,
+/* 112 */          NoSymbol,
+/* 113 */          NoSymbol,
+/* 114 */          NoSymbol,
+/* 115 */          NoSymbol,
+/* 116 */          NoSymbol,
+/* 117 */          NoSymbol,
+/* 118 */          NoSymbol,
+/* 119 */          NoSymbol,
+/* 120 */          NoSymbol,
+/* 121 */          NoSymbol,
+/* 122 */          NoSymbol,
+/* 123 */          NoSymbol,
+/* 124 */          NoSymbol,
+/* 125 */          NoSymbol,
+/* 126 */          NoSymbol,
+/* 127 */          NoSymbol,
+/* -- ShiftMask -- */
+/* 0 unused */     NoSymbol,
+/* 1 ESCape */     XK_Escape,
+/* 2 1 */          XK_exclam,
+/* 3 2 */          XK_at,
+/* 4 3 */          XK_numbersign,
+/* 5 4 */          XK_dollar,
+/* 6 5 */          XK_percent,
+/* 7 6 */          XK_asciicircum,
+/* 8 7 */          XK_ampersand,
+/* 9 8 */          XK_asterisk,
+/* 10 9 */         XK_parenleft,
+/* 11 0 */         XK_parenright,
+/* 12 - */         XK_underscore,
+/* 13 = */         XK_plus,
+/* 14 backspace */ XK_BackSpace,
+/* 15 tab */       XK_ISO_Left_Tab,
+/* 16 q */         XK_Q,
+/* 17 w */         XK_W,
+/* 18 e */         XK_E,
+/* 19 r */         XK_R,
+/* 20 t */         XK_T,
+/* 21 y */         XK_Y,
+/* 22 u */         XK_U,
+/* 23 i */         XK_I,
+/* 24 o */         XK_O,
+/* 25 p */         XK_P,
+/* 26 [ */         XK_braceleft,
+/* 27 ] */         XK_braceright,
+/* 28 return */    XK_Return,
+/* 29 control */   XK_Control_L,
+/* 30 a */         XK_A,
+/* 31 s */         XK_S,
+/* 32 d */         XK_D,
+/* 33 f */         XK_F,
+/* 34 g */         XK_G,
+/* 35 h */         XK_H,
+/* 36 j */         XK_J,
+/* 37 k */         XK_K,
+/* 38 l */         XK_L,
+/* 39 ; */         XK_colon,
+/* 40 ' */         XK_quotedbl,
+/* 41 ` */         XK_asciitilde,
+/* 42 shift */     XK_Shift_L,
+/* 43 \ */         XK_bar,
+/* 44 z */         XK_Z,
+/* 45 x */         XK_X,
+/* 46 c */         XK_C,
+/* 47 v */         XK_V,
+/* 48 b */         XK_B,
+/* 49 n */         XK_N,
+/* 50 m */         XK_M,
+/* 51 , */         XK_less,
+/* 52 . */         XK_greater,
+/* 53 / */         XK_question,
+/* 54 shift */     XK_Shift_R,
+/* 55 kp * */      NoSymbol, // TODO ?????
+/* 56 alt */       XK_Alt_L,
+/* 57 space */     XK_space,
+/* 58 caps */      XK_Caps_Lock,
+/* 59 f1 */        XK_F1,
+/* 60 f2 */        XK_F2,
+/* 61 f3 */        XK_F3,
+/* 62 f4 */        XK_F4,
+/* 63 f5 */        XK_F5,
+/* 64 f6 */        XK_F6,
+/* 65 f7 */        XK_F7,
+/* 66 f8 */        XK_F8,
+/* 67 f9 */        XK_F9,
+/* 68 f10 */       XK_F10,
+/* 69 num lock */  XK_Num_Lock,
+/* 70 scroll lock*/ XK_Scroll_Lock,
+/* 71 kp 7 */      NoSymbol,
+/* 72 kp 8 */      NoSymbol,
+/* 73 kp 9 */      NoSymbol,
+/* 74 kp - */      NoSymbol,
+/* 75 kp 4 */      NoSymbol,
+/* 76 kp 5 */      NoSymbol,
+/* 77 kp 6 */      NoSymbol,
+/* 78 kp + */      NoSymbol,
+/* 79 kp 1 */      NoSymbol,
+/* 80 kp 2 */      NoSymbol,
+/* 81 kp 3 */      NoSymbol,
+/* 82 kp 0 */      NoSymbol,
+/* 83 kp . */      NoSymbol,
+/* 84 0 */         NoSymbol,
+/* 85 0 */         NoSymbol,
+/* 86 0 */         NoSymbol,
+/* 87 f11 */       XK_F11,
+/* 88 f12 */       XK_F12,
+/* 89 0 */         NoSymbol,
+/* 90 0 */         NoSymbol,
+/* 91 0 */         XK_Super_L,
+/* 92 0 */         XK_Super_R,
 /* 93 0 */         NoSymbol,
 /* 94 0 */         NoSymbol,
 /* 95 0 */         NoSymbol,
@@ -139,25 +269,28 @@
 /* 124 */          NoSymbol,
 /* 125 */          NoSymbol,
 /* 126 */          NoSymbol,
-/* 127 */          NoSymbol
+/* 127 */          NoSymbol,
 };
 
 
 static void* event_watcher(void* arg) {
   printf("hello\n");
   int fd = *((int*) arg);
-	unsigned char c;
+
+	unsigned char buf[256], *b;
+  int n;
 
 	while (1) {
-    if (read(fd, &c, 1) > 0) {
-      KdEnqueueKeyboardEvent(c & 0x7f, c & 0x80);
+    while ((n = read(fd, buf, sizeof(buf))) > 0) {
+      b = buf;
+      while (n--) {
+        KdEnqueueKeyboardEvent(b[0] & 0x7f, b[0] & 0x80);
+        b++;
+      }
     }
-    usleep(200);
   }
 }
 
-#define NR_KEYS 101
-
 static void PhoenixKeyboardLoad(void)
 {
 	kdMinScanCode = 0;
@@ -165,13 +298,10 @@
 
   KeySym *k;
   int i;
-	int row;
-
-	row = 0;
-	for (i = 0; i < NR_KEYS && row < KD_MAX_LENGTH; ++i) {
-    k = kdKeymap + row * KD_MAX_WIDTH;
+	for (i = 0; i < kdMaxScanCode; ++i) {
+    k = kdKeymap + i * KD_MAX_WIDTH;
     k[0] = phoenix_to_x[i];
-    row++;
+    k[1] = phoenix_to_x[i+128];
   }
 }
 
unchanged:
--- a/kdrive/phoenix/phoenix.c	2024-08-19 10:04:19.200262718 +0200
+++ b/kdrive/phoenix/phoenix.c	2024-08-29 12:03:47.660542800 +0200
@@ -8,6 +8,10 @@
 #include <sys/ioctl.h>
 #include <X11/keysym.h>
 
+static void signalHandler(int signo)
+{
+}
+
 static int PhoenixInit(void)
 {
 	return 1;
@@ -15,6 +19,9 @@
 
 static void PhoenixEnable(void)
 {
+	signal(SIGINT, signalHandler);
+	signal(SIGQUIT, signalHandler);
+	signal(SIGTERM, signalHandler);
 }
 
 static Bool PhoenixSpecialKey(KeySym sym)
