diff -ruN a/kdrive/phoenix/phoenix.c b/kdrive/phoenix/phoenix.c
--- a/kdrive/phoenix/phoenix.c	2024-09-02 10:24:29.020264244 +0200
+++ b/kdrive/phoenix/phoenix.c	2024-09-02 11:36:26.806165111 +0200
@@ -4,6 +4,7 @@
 #include "kdrive.h"
 #include <errno.h>
 #include <signal.h>
+#include <termios.h>
 #include <sys/stat.h>
 #include <sys/ioctl.h>
 #include <X11/keysym.h>
@@ -14,6 +15,17 @@
 
 static int PhoenixInit(void)
 {
+  int fd;
+  if ((fd = open("/dev/tty0", O_WRONLY, 0)) < 0) {
+    FatalError("PhoenixInit: Cannot open /dev/tty0 (%s)\n",
+         strerror(errno));
+  }
+
+	if (ioctl(fd, TIOCNOTTY, NULL) < 0) {
+		FatalError("PhoenixInit: TIOCNOTTY failed\n");
+	}
+  close(fd);
+
 	return 1;
 }
 
