--- fs_mark-3.3/fs_mark.h	2023-07-10 15:47:05.889696230 +0200
+++ fs_mark-3.3.phoenix/fs_mark.h	2023-11-21 17:46:54.622739540 +0100
@@ -18,14 +18,21 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
+/*
+ * Define used in targets running with dummyfs
+ */
+#undef DUMMYFS_TARGET
+#if defined(__CPU_IMXRT106X) || defined(__CPU_IMXRT117X) || defined(__CPU_STM32L4X6)
+#define DUMMYFS_TARGET
+#endif
 
 /*
  * Default and maximum parameters.
  */
-#define MAX_IO_BUFFER_SIZE 	(1024 * 1024) 	/* Max write buffer size is 1MB */
+#define MAX_IO_BUFFER_SIZE 	(1024) 	/* Max write buffer size is 1KB */
 #define MAX_FILES		(1000000)	/* Max number of files to test of each size */
 #define MAX_THREADS		(64)		/* Max number of threads allowed */
-#define MAX_NAME_PATH		(1000)		/* Length of the pathname before the leaf */
+#define MAX_NAME_PATH		(200)		/* Length of the pathname before the leaf */
 #define FILENAME_SIZE		(128) 		/* Max length of filenames */
 #define MAX_STRING_SIZE		(160)	    	/* Max number of bytes in a string */
 
@@ -34,7 +41,7 @@
  */
 #define DEFAULT_SECS_PER_DIR	(180)
 #define DEFAULT_FILE_SIZE	(50 * 1024)
-#define DEFAULT_IO_SIZE		(16 * 1024)
+#define DEFAULT_IO_SIZE		(1024)
 #define DEFAULT_NUM_FILES	(1000)
 #define DEFAULT_NAME_LEN	(40)
 #define DEFAULT_RAND_NAME	(24)
@@ -91,6 +98,9 @@
 	"SYNC POST: Issue sync() and then reopen and fsync() each file in order after main write loop."
 };
 
+#ifdef DUMMYFS_TARGET
+unsigned int dummyfs_size = 0;
+#endif
 
 /*
  * Use the normal fsync() per file by default
@@ -125,7 +135,7 @@
 int	num_threads = 1;			/* Number of threads */
 int	do_fill_fs = 0;				/* Run until the file system is full  */
 int	verbose_stats = 0;		    	/* Print complete stats for each system call */
-char 	log_file_name[PATH_MAX] = "fs_log.txt"; /* Log file name for run */
+char 	log_file_name[PATH_MAX - 16] = "fs_log.txt"; /* Log file name for run - reserve place for suffix */
 FILE	*log_file_fp;				/* Parent file pointer for log file  */
 FILE	*child_log_file_fp;			/* Child file pointer for log file  */
 
