diff -Naur a/sysklogd/syslogd.c b/sysklogd/syslogd.c
--- a/sysklogd/syslogd.c	2021-12-16 23:27:24.952624736 +0100
+++ b/sysklogd/syslogd.c	2021-12-21 12:53:44.699561975 +0100
@@ -950,6 +950,7 @@
 	unlink(sunx.sun_path);
 
 	sock_fd = xsocket(AF_UNIX, SOCK_DGRAM, 0);
+	setsockopt_int(sock_fd, SOL_SOCKET, SO_RCVBUF, 16384);
 	xbind(sock_fd, (struct sockaddr *) &sunx, sizeof(sunx));
 	chmod(_PATH_LOG, 0666);
 
