--- busybox-1.27.2/coreutils/ls.c	2024-08-22 17:07:54.832370546 +0200
+++ busybox-1.27.2.phx/coreutils/ls.c	2024-08-27 16:56:15.357812323 +0200
@@ -163,6 +163,9 @@
 //usage:     "\n	--color[={always,never,auto}]	Control coloring"
 //usage:	)
 
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <unistd.h>
 #include "libbb.h"
 #include "common_bufsiz.h"
 #include "unicode.h"
@@ -927,6 +930,7 @@
 	}
 	dn = NULL;
 	nfiles = 0;
+
 	while ((entry = readdir(dir)) != NULL) {
 		char *fullname;
 
@@ -943,6 +947,7 @@
 		fullname = concat_path_file(path, entry->d_name);
 		cur = my_stat(fullname, bb_basename(fullname), 0);
 		if (!cur) {
+			printf("Error: %s\n", fullname);
 			free(fullname);
 			continue;
 		}
