diff -ruN c/compat/imsg.c d/compat/imsg.c
--- c/compat/imsg.c	2022-02-03 11:08:54.689945627 +0100
+++ d/compat/imsg.c	2022-02-03 11:29:35.205311722 +0100
@@ -93,7 +93,12 @@
 again:
 	if (getdtablecount() + imsg_fd_overhead +
 	    (int)((CMSG_SPACE(sizeof(int))-CMSG_SPACE(0))/sizeof(int))
+#ifdef __phoenix__
+	    /* getdtablesize is not supported in libphoenix (non-POSIX) */
+	    >= sysconf(_SC_OPEN_MAX)) {
+#else
 	    >= getdtablesize()) {
+#endif
 		errno = EAGAIN;
 		free(ifd);
 		return (-1);
