diff -ruN a/compat/setproctitle.c b/compat/setproctitle.c
--- a/compat/setproctitle.c	2021-08-18 22:06:04.199290361 +0200
+++ b/compat/setproctitle.c	2021-08-09 08:49:35.379973445 +0200
@@ -128,7 +128,7 @@
 	char buf[1024], ptitle[1024];
 	size_t len = 0;
 	int r;
-	extern char *__progname;
+	const char *progname = "iked";
 #if SPT_TYPE == SPT_PSTAT
 	union pstun pst;
 #endif
@@ -138,7 +138,7 @@
 		return;
 #endif
 
-	strlcpy(buf, __progname, sizeof(buf));
+	strlcpy(buf, progname, sizeof(buf));
 
 	r = -1;
 	va_start(ap, fmt);
diff -ruN a/ikectl/ikectl.c b/ikectl/ikectl.c
--- a/ikectl/ikectl.c	2021-08-18 22:06:05.779062151 +0200
+++ b/ikectl/ikectl.c	2021-08-09 08:49:35.379973445 +0200
@@ -71,10 +71,10 @@
 __dead void
 usage(void)
 {
-	extern char *__progname;
+	const char *progname = "ikectl";
 
 	fprintf(stderr, "usage: %s [-q] [-s socket] command [arg ...]\n",
-	    __progname);
+	    progname);
 	exit(1);
 }
 
diff -ruN a/iked/iked.c b/iked/iked.c
--- a/iked/iked.c	2021-06-11 10:54:25.744913258 +0200
+++ b/iked/iked.c	2021-06-23 11:51:49.898253943 +0200
@@ -60,10 +60,10 @@
 __dead void
 usage(void)
 {
-	extern char	*__progname;
+	const char *progname = "iked";
 
 	fprintf(stderr, "usage: %s [-dnSTtv] [-D macro=value] "
-	    "[-f file] [-p udpencap_port] [-s socket]\n", __progname);
+	    "[-f file] [-p udpencap_port] [-s socket]\n", progname);
 	exit(1);
 }
 
diff -ruN a/iked/log.c b/iked/log.c
--- a/iked/log.c	2021-08-18 22:06:06.654935624 +0200
+++ b/iked/log.c	2021-08-23 14:01:42.952012928 +0200
@@ -52,14 +52,14 @@
 void
 log_init(int n_debug, int facility)
 {
-	extern char	*__progname;
+	const char *progname = "iked";
 
 	debug = n_debug;
 	verbose = n_debug;
-	log_procinit(__progname);
+	log_procinit(progname);
 
 	if (!debug)
-		openlog(__progname, LOG_PID | LOG_NDELAY, facility);
+		openlog(progname, LOG_PID | LOG_NDELAY, facility);
 
 	tzset();
 }
diff -ruN a/regress/test_helper/test_helper.c b/regress/test_helper/test_helper.c
--- a/regress/test_helper/test_helper.c	2021-06-11 10:55:18.790954990 +0200
+++ b/regress/test_helper/test_helper.c	2021-06-11 17:46:31.576973750 +0200
@@ -96,7 +100,7 @@
 		}				\
 	} while (0)
 
-extern char *__progname;
+static char *progname = "test_helper";
 
 static int verbose_mode = 0;
 static int quiet_mode = 0;
@@ -126,13 +130,13 @@
 			break;
 		default:
 			fprintf(stderr, "Unrecognised command line option\n");
-			fprintf(stderr, "Usage: %s [-v]\n", __progname);
+			fprintf(stderr, "Usage: %s [-v]\n", progname);
 			exit(1);
 		}
 	}
 	setvbuf(stdout, NULL, _IONBF, 0);
 	if (!quiet_mode)
-		printf("%s: ", __progname);
+		printf("%s: ", progname);
 	if (verbose_mode)
 		printf("\n");
