diff -ruN a/iked/iked.h b/iked/iked.h
--- a/iked/iked.h   2021-06-11 10:54:25.776962971 +0200
+++ b/iked/iked.h   2021-06-11 13:29:56.682142879 +0200
@@ -815,6 +815,9 @@
 	unsigned int		  nxfs;
 };
 
+#define roundup(x, y) ((((x) + ((y) -1)) / (y)) * (y))
+#define MIN(a, b) (((a) < (b)) ? (a) : (b))
+
 /* iked.c */
 void	 parent_reload(struct iked *, int, const char *);
 
diff -ruN a/regress/test_helper/test_helper.c b/regress/test_helper/test_helper.c
--- a/regress/test_helper/test_helper.c 2021-06-11 10:55:18.790954990 +0200
+++ b/regress/test_helper/test_helper.c 2021-06-11 17:46:31.576973750 +0200
@@ -34,6 +34,8 @@
 
 #include "test_helper.h"
 
+#define MIN(a,b) (((a)<(b))?(a):(b))
+
 #define TEST_CHECK_INT(r, pred) do {		\
 		switch (pred) {			\
 		case TEST_EQ:			\
