diff -ruN c/compat/arc4random_linux.h d/compat/arc4random_linux.h
--- c/compat/arc4random_linux.h	2022-02-03 13:10:38.907653741 +0100
+++ d/compat/arc4random_linux.h	2022-02-03 13:12:05.026537928 +0100
@@ -76,6 +76,17 @@
 static inline int
 _rs_allocate(struct _rs **rsp, struct _rsx **rsxp)
 {
+#ifdef __phoenix__
+	/* libphoenix mmap is not posix compliant */
+	if ((*rsp = malloc(sizeof(**rsp))) == NULL)
+		return (-1);
+
+	if ((*rsxp = malloc(sizeof(**rsxp))) == NULL) {
+		free(*rsp);
+		*rsp = NULL;
+		return (-1);
+	}
+#else
 	if ((*rsp = mmap(NULL, sizeof(**rsp), PROT_READ|PROT_WRITE,
 	    MAP_ANON|MAP_PRIVATE, -1, 0)) == MAP_FAILED)
 		return (-1);
@@ -86,6 +97,7 @@
 		*rsp = NULL;
 		return (-1);
 	}
+#endif
 
 	_ARC4_ATFORK(_rs_forkhandler);
 	return (0);
