diff -ruN a/compat/arc4random_linux.h b/compat/arc4random_linux.h
--- a/compat/arc4random_linux.h	2021-08-18 22:06:04.191291517 +0200
+++ b/compat/arc4random_linux.h	2021-08-23 12:45:40.550667256 +0200
@@ -37,8 +37,12 @@
 extern int __register_atfork(void (*)(void), void(*)(void), void (*)(void), void *);
 #define _ARC4_ATFORK(f) __register_atfork(NULL, NULL, (f), __dso_handle)
 #else
+#ifdef __phoenix__
+#define _ARC4_ATFORK(f) /* pthread_atfork is not implemented in libphoenix */
+#else
 #define _ARC4_ATFORK(f) pthread_atfork(NULL, NULL, (f))
 #endif
+#endif
 
 static inline void
 _getentropy_fail(void)

