diff -ruN c/iked/iked.c d/iked/iked.c
--- c/iked/iked.c	2022-02-03 13:09:44.858787336 +0100
+++ d/iked/iked.c	2022-02-03 15:46:00.871960359 +0100
@@ -33,6 +33,7 @@
 #include <err.h>
 #include <pwd.h>
 #include <event.h>
+#include <fcntl.h>
 
 #include "iked.h"
 #include "ikev2.h"
@@ -57,6 +58,49 @@
 	{ "ikev2",	PROC_IKEV2,	parent_dispatch_ikev2, ikev2 }
 };
 
+#ifdef __phoenix__
+/* daemon is not supported in libphoenix (non-POSIX) */
+static int
+daemon(int nochdir, int noclose)
+{
+	pid_t pid = fork();
+
+	if (pid == -1)
+		return -1;
+
+	if (pid != 0)
+		exit(0);
+
+	if (nochdir == 0) {
+		if (chdir("/") < 0)
+			return -1;
+	}
+
+	if (noclose == 0) {
+		int devnull = open("/dev/null", O_RDWR);
+		if (devnull < 0)
+			return -1;
+
+		if (dup2(devnull, STDIN_FILENO) < 0) {
+			close(devnull);
+			return -1;
+		}
+
+		if (dup2(devnull, STDOUT_FILENO) < 0) {
+			close(devnull);
+			return -1;
+		}
+
+		if (dup2(devnull, STDERR_FILENO) < 0) {
+			close(devnull);
+			return -1;
+		}
+	}
+
+	return 0;
+}
+#endif
+
 __dead void
 usage(void)
 {
