diff -ruN a/ikectl/ikeca.c b/ikectl/ikeca.c
--- a/ikectl/ikeca.c	2021-08-23 14:54:15.149410633 +0200
+++ b/ikectl/ikeca.c	2021-08-18 22:25:45.367650782 +0200
@@ -912,8 +912,12 @@
 			    NULL };
 			ca_execv(cmd);
 		}
+#ifdef __phoenix__
+		printf("exported files in %s\n", src);
+#else
 		if (realpath(src, dst) != NULL)
 			printf("exported files in %s\n", dst);
+#endif
 	}
 
 	if (stat(PATH_ZIP, &st) == 0) {

