diff -ruN x/iked/ikev2.c y/iked/ikev2.c
--- a/iked/ikev2.c	2021-06-25 16:49:06.269105404 +0200
+++ b/iked/ikev2.c	2021-06-25 17:01:14.148001910 +0200
@@ -605,7 +605,7 @@
 
 	logit(hdr->ike_exchange == IKEV2_EXCHANGE_INFORMATIONAL ?
 	    LOG_DEBUG : LOG_INFO,
-	    "%srecv %s %s %u peer %s local %s, %ld bytes, policy '%s'",
+	    "%srecv %s %s %u peer %s local %s, %zu bytes, policy '%s'",
 	    SPI_IH(hdr),
 	    print_map(hdr->ike_exchange, ikev2_exchange_map),
 	    msg->msg_response ? "res" : "req",
diff -ruN x/iked/ikev2_msg.c y/iked/ikev2_msg.c
--- a/iked/ikev2_msg.c	2021-04-15 19:18:01.000000000 +0200
+++ b/iked/ikev2_msg.c	2021-06-25 17:02:07.120391473 +0200
@@ -271,7 +271,7 @@
 	exchange = hdr->ike_exchange;
 	flags = hdr->ike_flags;
 	logit(exchange == IKEV2_EXCHANGE_INFORMATIONAL ?  LOG_DEBUG : LOG_INFO,
-	    "%ssend %s %s %u peer %s local %s, %ld bytes%s",
+	    "%ssend %s %s %u peer %s local %s, %zu bytes%s",
 	    SPI_IH(hdr),
 	    print_map(exchange, ikev2_exchange_map),
 	    (flags & IKEV2_FLAG_RESPONSE) ? "res" : "req",
diff -ruN x/iked/ikev2_pld.c y/iked/ikev2_pld.c
--- a/iked/ikev2_pld.c	2021-06-25 16:57:01.202051869 +0200
+++ b/iked/ikev2_pld.c	2021-06-25 16:57:59.178516721 +0200
@@ -319,7 +319,7 @@
 	 */
 	if (!sap->sap_more && left != sap_length) {
 		log_debug("%s: malformed payload: SA payload length mismatches "
-		    "single proposal substructure length (%lu != %zu)",
+		    "single proposal substructure length (%zu != %zu)",
 		    __func__, left, sap_length);
 		return (-1);
 	}
