diff -ruN a/iked/util.c b/iked/util.c
--- a/iked/util.c	2021-08-23 16:27:21.913923321 +0200
+++ b/iked/util.c	2021-08-23 14:01:46.123524262 +0200
@@ -167,10 +167,12 @@
 		    IPPROTO_IP,
 		    IP_IPSEC_POLICY
 	};
+#ifndef __phoenix__
 	int	 a6[] = {
 		    IPPROTO_IPV6,
 		    IPV6_IPSEC_POLICY,
 	};
+#endif
 	struct sadb_x_policy pol = {
 		    SADB_UPDATE,
 		    SADB_EXT_SENSITIVITY,
@@ -182,9 +184,11 @@
 	case AF_INET:
 		a = a4;
 		break;
+#ifndef __phoenix__
 	case AF_INET6:
 		a = a6;
 		break;
+#endif
 	default:
 		log_warn("%s: invalid address family", __func__);
 		return (-1);
diff -ruN a/iked/iked.c b/iked/iked.c
--- a/iked/iked.c	2021-08-23 17:19:20.512777989 +0200
+++ b/iked/iked.c	2021-08-23 17:30:01.681726638 +0200
@@ -310,6 +310,7 @@
 	if (env->sc_nattmode != NATT_DISABLE)
 		config_setsocket(env, &ss, htons(env->sc_nattport), PROC_IKEV2, 1);
 
+#ifndef __phoenix__
 	bzero(&ss, sizeof(ss));
 	ss.ss_family = AF_INET6;
 
@@ -317,6 +318,7 @@
 		config_setsocket(env, &ss, htons(IKED_IKE_PORT), PROC_IKEV2, 0);
 	if (env->sc_nattmode != NATT_DISABLE)
 		config_setsocket(env, &ss, htons(env->sc_nattport), PROC_IKEV2, 1);
+#endif
 
 	/*
 	 * pledge in the parent process:

