diff -ruN a/iked/util.c b/iked/util.c
--- a/iked/util.c	2021-08-23 16:36:36.260953176 +0200
+++ b/iked/util.c	2021-08-23 14:01:46.123524262 +0200
@@ -232,11 +232,13 @@
 		goto bad;
 	}
 
+#ifndef __phoenix__
 	val = 1;
 	if (setsockopt(s, SOL_SOCKET, SO_REUSEPORT, &val, sizeof(int)) == -1) {
 		log_warn("%s: failed to set reuseport", __func__);
 		goto bad;
 	}
+#endif /* __phoenix__ */
 	val = 1;
 	if (setsockopt(s, SOL_SOCKET, SO_REUSEADDR, &val, sizeof(int)) == -1) {
 		log_warn("%s: failed to set reuseaddr", __func__);

