diff -ruN a/compat/imsg-buffer.c b/compat/imsg-buffer.c
--- a/compat/imsg-buffer.c	2021-08-23 17:10:41.962801047 +0200
+++ b/compat/imsg-buffer.c	2021-08-23 12:48:27.519210651 +0200
@@ -183,7 +183,11 @@
 int
 ibuf_write(struct msgbuf *msgbuf)
 {
+#ifdef __phoenix__
+	struct iovec	 iov[32]; /* avoid stack overflow */
+#else
 	struct iovec	 iov[IOV_MAX];
+#endif
 	struct ibuf	*buf;
 	unsigned int	 i = 0;
 	ssize_t	n;

