diff -ruN a/compat/openbsd-compat.h b/compat/openbsd-compat.h
--- a/compat/openbsd-compat.h	2021-04-15 19:18:01.000000000 +0200
+++ b/compat/openbsd-compat.h	2021-08-20 13:53:14.076987280 +0200
@@ -238,8 +238,8 @@
 #define CPI_PRIVATE_MAX		65535
 #endif
 
-#if !defined(SADB_X_ADDFLOW) && defined(SADB_X_SPDUPDATE)
-#define SADB_X_ADDFLOW	SADB_X_SPDUPDATE
+#if !defined(SADB_X_ADDFLOW) && defined(SADB_X_SPDADD)
+#define SADB_X_ADDFLOW	SADB_X_SPDADD
 #endif
 #if !defined(SADB_X_DELFLOW) && defined(SADB_X_SPDDELETE)
 #define SADB_X_DELFLOW	SADB_X_SPDDELETE

