diff -ruN a/iked/iked.h b/iked/iked.h
--- a/iked/iked.h	2021-08-30 18:14:40.316135606 +0200
+++ b/iked/iked.h	2021-08-30 21:56:41.435001459 +0200
@@ -1106,7 +1106,7 @@
 int	 pfkey_sa_update_addresses(int, struct iked_childsa *);
 int	 pfkey_sa_delete(int, struct iked_childsa *);
 int	 pfkey_sa_last_used(int, struct iked_childsa *, uint64_t *);
-int	 pfkey_flush(int);
+int	 pfkey_flush(int, uint8_t);
 int	 pfkey_socket(void);
 void	 pfkey_init(struct iked *, int fd);
 
diff -ruN a/iked/pfkey.c b/iked/pfkey.c
--- a/iked/pfkey.c	2021-08-30 18:14:40.320135728 +0200
+++ b/iked/pfkey.c	2021-08-30 21:58:56.901828983 +0200
@@ -1939,7 +1939,7 @@
 }
 
 int
-pfkey_flush(int sd)
+pfkey_flush(int sd, uint8_t what)
 {
 	struct sadb_msg smsg;
 	struct iovec	iov[IOV_CNT];
@@ -1950,7 +1950,7 @@
 	smsg.sadb_msg_seq = ++sadb_msg_seq;
 	smsg.sadb_msg_pid = getpid();
 	smsg.sadb_msg_len = sizeof(smsg) / 8;
-	smsg.sadb_msg_type = SADB_FLUSH;
+	smsg.sadb_msg_type = what;
 	smsg.sadb_msg_satype = SADB_SATYPE_UNSPEC;
 
 	iov_cnt = 0;
@@ -2022,7 +2022,8 @@
 	if ((fd = socket(PF_KEY, SOCK_RAW, PF_KEY_V2)) == -1)
 		fatal("pfkey_socket: failed to open PF_KEY socket");
 
-	pfkey_flush(fd);
+	pfkey_flush(fd, SADB_FLUSH);
+	pfkey_flush(fd, SADB_X_SPDFLUSH);
 
 	return (fd);
 }
