diff -ruN c/iked/iked.h d/iked/iked.h
--- c/iked/iked.h	2022-02-03 17:37:38.209176404 +0100
+++ d/iked/iked.h	2022-02-03 17:53:52.557616454 +0100
@@ -1106,7 +1106,7 @@
 int	 pfkey_sa_update_addresses(int, struct iked_childsa *);
 int	 pfkey_sa_delete(int, struct iked_childsa *);
 int	 pfkey_sa_last_used(int, struct iked_childsa *, uint64_t *);
-int	 pfkey_flush(int);
+int	 pfkey_flush(int, uint8_t);
 int	 pfkey_socket(void);
 void	 pfkey_init(struct iked *, int fd);
 
diff -ruN c/iked/pfkey.c d/iked/pfkey.c
--- c/iked/pfkey.c	2022-02-03 17:48:15.787914144 +0100
+++ d/iked/pfkey.c	2022-02-03 17:53:52.557616454 +0100
@@ -1940,7 +1940,7 @@
 }
 
 int
-pfkey_flush(int sd)
+pfkey_flush(int sd, uint8_t what)
 {
 	struct sadb_msg smsg;
 	struct iovec	iov[IOV_CNT];
@@ -1951,7 +1951,7 @@
 	smsg.sadb_msg_seq = ++sadb_msg_seq;
 	smsg.sadb_msg_pid = getpid();
 	smsg.sadb_msg_len = sizeof(smsg) / 8;
-	smsg.sadb_msg_type = SADB_FLUSH;
+	smsg.sadb_msg_type = what;
 	smsg.sadb_msg_satype = SADB_SATYPE_UNSPEC;
 
 	iov_cnt = 0;
@@ -2023,7 +2023,8 @@
 	if ((fd = socket(PF_KEY, SOCK_RAW, PF_KEY_V2)) == -1)
 		fatal("pfkey_socket: failed to open PF_KEY socket");
 
-	pfkey_flush(fd);
+	pfkey_flush(fd, SADB_FLUSH);
+	pfkey_flush(fd, SADB_X_SPDFLUSH);
 
 	return (fd);
 }
