diff -ruN a/iked/ikev2.c b/iked/ikev2.c
--- a/iked/ikev2.c	2021-08-30 18:14:40.316135606 +0200
+++ b/iked/ikev2.c	2021-08-30 21:56:42.331949643 +0200
@@ -6515,11 +6515,14 @@
 {
 	struct iked_flow	*flow;
 	struct iked_sa		*sa;
+#if 0
 	struct iked_policy	 pol, *p = NULL;
+#endif
 
 	if (env->sc_passive)
 		return (0);
 
+#if 0
 	/* First try to find an active flow with IKE SA */
 	flow = RB_FIND(iked_flows, &env->sc_activeflows, acquire);
 	if (!flow) {
@@ -6547,6 +6550,8 @@
 			    "IKE_SA_INIT exchange for policy '%s'",
 			    __func__, p->pol_name);
 	} else {
+#endif
+	if ((flow = acquire) != NULL) {
 		log_debug("%s: found active flow", __func__);
 
 		if ((sa = flow->flow_ikesa) == NULL) {
