diff -ruN c/iked/pfkey.c d/iked/pfkey.c
--- c/iked/pfkey.c	2022-02-03 17:54:45.547200957 +0100
+++ d/iked/pfkey.c	2022-02-03 17:57:31.024218888 +0100
@@ -1184,6 +1184,10 @@
 int
 pfkey_sa_lookup(int sd, struct iked_childsa *sa, uint64_t *last_used)
 {
+#ifdef __phoenix__
+	/* SADB_GET is not supported by our ipsec */
+	return (-1);
+#else
 	struct sadb_msg		*msg, smsg;
 	struct sadb_address	 sa_src, sa_dst;
 	struct sadb_sa		 sadb;
@@ -1340,6 +1344,7 @@
 done:
 	freezero(data, n);
 	return (ret);
+#endif
 }
 
 int
