diff -ruN a/iked/pfkey.c b/iked/pfkey.c
--- a/iked/pfkey.c	2021-08-30 22:19:17.730765767 +0200
+++ b/iked/pfkey.c	2021-08-30 22:17:52.844310482 +0200
@@ -1184,6 +1184,7 @@
 int
 pfkey_sa_lookup(int sd, struct iked_childsa *sa, uint64_t *last_used)
 {
+#if 0
 	struct sadb_msg		*msg, smsg;
 	struct sadb_address	 sa_src, sa_dst;
 	struct sadb_sa		 sadb;
@@ -1340,6 +1341,9 @@
 done:
 	freezero(data, n);
 	return (ret);
+#else
+	return (-1); /* SADB_GET is not supported */
+#endif
 }
 
 int
