diff -ruN c/iked/iked.c d/iked/iked.c
--- c/iked/iked.c	2022-02-03 16:16:27.025160211 +0100
+++ d/iked/iked.c	2022-02-04 15:15:04.579849902 +0100
@@ -54,7 +54,15 @@
 
 static struct privsep_proc procs[] = {
 	{ "ca",		PROC_CERT,	parent_dispatch_ca, caproc, IKED_CA },
+#ifdef __phoenix__
+	/*
+	 * control process blocks on accept4 call (kernel doesn't support non-blocking call)
+	 * so it's impossible to kill it and stop iked, disable control process for now as
+	 * we don't use ikectl anyway
+	 */
+#else
 	{ "control",	PROC_CONTROL,	parent_dispatch_control, control },
+#endif
 	{ "ikev2",	PROC_IKEV2,	parent_dispatch_ikev2, ikev2 }
 };
 
diff -ruN c/iked/ikev2.c d/iked/ikev2.c
--- c/iked/ikev2.c	2022-02-03 17:37:38.213176469 +0100
+++ d/iked/ikev2.c	2022-02-04 15:15:04.583849456 +0100
@@ -190,7 +190,15 @@
 static struct privsep_proc procs[] = {
 	{ "parent",	PROC_PARENT,	ikev2_dispatch_parent },
 	{ "certstore",	PROC_CERT,	ikev2_dispatch_cert },
+#ifdef __phoenix__
+	/*
+	 * control process blocks on accept4 call (kernel doesn't support non-blocking call)
+	 * so it's impossible to kill it and stop iked, disable control process for now as
+	 * we don't use ikectl anyway
+	 */
+#else
 	{ "control",	PROC_CONTROL,	ikev2_dispatch_control }
+#endif
 };
 
 pid_t
