diff -ruN a/compat/imsg.c b/compat/imsg.c
--- a/compat/imsg.c	2021-08-30 22:19:17.622804314 +0200
+++ b/compat/imsg.c	2022-01-19 16:47:47.722393075 +0100
@@ -70,7 +70,16 @@
 	struct cmsghdr		*cmsg;
 	union {
 		struct cmsghdr hdr;
+#ifdef __phoenix__
+		/*
+		 * Accept more than one file descriptor,
+		 * this is specific to kernel implementation
+		 * of file descriptor passing in the kernel.
+		 */
+		char	buf[CMSG_SPACE(sizeof(int) * 4)];
+#else
 		char	buf[CMSG_SPACE(sizeof(int) * 1)];
+#endif
 	} cmsgbuf;
 	struct iovec		 iov;
 	ssize_t			 n = -1;
@@ -131,7 +140,16 @@
 					ifd->fd = fd;
 					TAILQ_INSERT_TAIL(&ibuf->fds, ifd,
 					    entry);
+#ifdef __phoenix__
+					/*
+					 * Accept more than one file descriptor,
+					 * this is specific to kernel implementation
+					 * of file descriptor passing in the kernel.
+					 */
+					ifd = calloc(1, sizeof(struct imsg_fd));
+#else
 					ifd = NULL;
+#endif
 				} else
 					close(fd);
 			}
