diff -ruN a/iked/iked.h b/iked/iked.h
--- a/iked/iked.h	2022-01-24 11:59:52.791501003 +0100
+++ b/iked/iked.h	2022-01-24 11:58:39.839699791 +0100
@@ -163,6 +163,7 @@
 
 	uint8_t				 flow_saproto;
 	uint8_t				 flow_ipproto;
+	uint32_t			 flow_id;
 
 	struct iked_addr		*flow_local;	/* outer source */
 	struct iked_addr		*flow_peer;	/* outer dest */
diff -ruN a/iked/pfkey.c b/iked/pfkey.c
--- a/iked/pfkey.c	2022-01-24 10:46:00.475836661 +0100
+++ b/iked/pfkey.c	2022-01-24 10:44:50.301914285 +0100
@@ -583,6 +583,8 @@
 	bzero(&sa_policy, sizeof(sa_policy));
 	sa_policy.sadb_x_policy_exttype = SADB_X_EXT_POLICY;
 	sa_policy.sadb_x_policy_dir = flow->flow_dir;
+	if (action == SADB_X_DELFLOW)
+		sa_policy.sadb_x_policy_id = flow->flow_id;
 
 	sa_policy.sadb_x_policy_type = IPSEC_POLICY_IPSEC;
 	bzero(&sa_ipsec, sizeof(sa_ipsec));
@@ -673,6 +675,9 @@
 		goto done;
 	}
 
+	flow->flow_id = sa_polid->sadb_x_policy_id;
+	log_debug("%s: flow with policy id 0x%x", __func__, flow->flow_id);
+
 	ret = 0;
 
  done:
