--- fs_mark-3.3/fs_mark.h	2023-07-10 15:47:05.889696230 +0200
+++ fs_mark-3.3.phoenix/fs_mark.h	2023-08-01 12:19:29.380631828 +0200
@@ -125,7 +125,7 @@
 int	num_threads = 1;			/* Number of threads */
 int	do_fill_fs = 0;				/* Run until the file system is full  */
 int	verbose_stats = 0;		    	/* Print complete stats for each system call */
-char 	log_file_name[PATH_MAX] = "fs_log.txt"; /* Log file name for run */
+char 	log_file_name[PATH_MAX - 16] = "fs_log.txt"; /* Log file name for run - reserve place for suffix */
 FILE	*log_file_fp;				/* Parent file pointer for log file  */
 FILE	*child_log_file_fp;			/* Child file pointer for log file  */
