--- a/core/iwasm/common/wasm_memory.c
+++ b/core/iwasm/common/wasm_memory.c
@@ -1501,8 +1501,9 @@
         new_mem = os_mremap(mapped_mem, old_size, new_size);
     }
     else {
-        new_mem = os_mmap(NULL, new_size, MMAP_PROT_NONE, MMAP_MAP_NONE,
-                          os_get_invalid_handle());
+        /* mprotect fix */
+        new_mem = os_mmap(NULL, new_size, MMAP_PROT_READ | MMAP_PROT_WRITE, MMAP_MAP_NONE,
+                          os_get_invalid_handle()); 
     }
     if (!new_mem) {
         return NULL;
@@ -1512,13 +1513,15 @@
     if (commit_size > 0
         && !os_mem_commit(new_mem, commit_size,
                           MMAP_PROT_READ | MMAP_PROT_WRITE)) {
-        os_munmap(new_mem, new_size);
+        os_munmap(new_mem, new_size);   
         return NULL;
     }
 #endif
 
-    if (os_mprotect(new_mem, commit_size, MMAP_PROT_READ | MMAP_PROT_WRITE)
-        != 0) {
+    /* mprotect fix */
+
+    if (os_mprotect(new_mem, new_size, MMAP_PROT_NONE) != 0 ||
+        os_mprotect(new_mem, commit_size, MMAP_PROT_READ | MMAP_PROT_WRITE) != 0) {
         wasm_munmap_linear_memory(new_mem, new_size, new_size);
         return NULL;
     }

