diff -ruN a/libtinyX11/Makefile b/libtinyX11/Makefile
--- a/libtinyX11/Makefile	2014-11-08 16:38:44.000000000 +0100
+++ b/libtinyX11/Makefile	2024-07-12 10:34:19.235959177 +0200
@@ -16,11 +16,6 @@
 
 all:	$(OBJS:.c=.o) 
 	ar cr libX11.a $(OBJS:.c=.o) $(XAUOBJS:.c=.o)
-	$(CC) $(LDFLAGS) -o libX11.so.6.2~ -shared -Wl,-soname,libX11.so.6.2 $(OBJS:.c=.o) $(XAUOBJS:.c=.o) \
-	-L../libXext -lXext
-	mv -f libX11.so.6.2~ libX11.so.6.2
-	ln -sf libX11.so.6.2 libX11.so.6
-	ln -sf libX11.so.6.2 libX11.so
 	
 	@echo "prefix=$(PREDIR)" > x11.pc
 	@echo "exec_prefix=$(PREDIR)/bin" >> x11.pc
@@ -53,9 +48,6 @@
 	
 install:
 	cp -af libX11.a $(DESTDIR)$(LIBDIR)
-	cp -af libX11.so.6.2 $(DESTDIR)$(LIBDIR)
-	cp -af libX11.so.6 $(DESTDIR)$(LIBDIR)
-	cp -af libX11.so $(DESTDIR)$(LIBDIR)
 	cp -a XKeysymDB $(DESTDIR)$(KEYSYMDB)
 	
 depend:
